# PHAGE FAMILY CLASSIFICATION UNDER Caudoviricetes: A REVIEW OF CURRENT TOOLS USING THE LATEST ICTV CLASSIFICATION FRAMEWORK  

# Yilin Zhu  

Dept. of Electrical Engineering City University of Hong Kong Kowloon, Hong Kong SAR, China zyl666777@gmail.com  

# Jiayu Shang  

Dept. of Electrical Engineering City University of Hong Kong Kowloon, Hong Kong SAR, China jyshang2-c@my.cityu.edu.hk  

# Cheng Peng  

Dept. of Electrical Engineering City University of Hong Kong Kowloon, Hong Kong SAR, China cpeng29-c@my.cityu.edu.hk  

# Yanni Sun  

Dept. of Electrical Engineering City University of Hong Kong Kowloon, Hong Kong SAR, China yannisun@cityu.edu.hk  

November 24, 2022  

# ABSTRACT  

Bacteriophages, which are viruses infecting bacteria, are the most ubiquitous and diverse entities in the biosphere. There is accumulating evidence revealing their important roles in shaping the structure of various microbiomes. Thanks to (viral) metagenomic sequencing, a large number of new bacteriophages have been discovered. However, lacking a standard and automatic virus classification pipeline, the taxonomic characterization of new viruses seriously lag behind the sequencing efforts. In particular, according to the latest version of ICTV, several large phage families in the previous classification system are removed. Therefore, a comprehensive review and comparison of taxonomic classification tools under the new standard are needed to establish the state-of-the-art. In this work, we retrained and tested four recently published tools on newly labeled databases. We demonstrated their utilities and tested them on multiple datasets, including the RefSeq, short contigs, simulated metagenomic datasets, and low-similarity datasets. This study provides a comprehensive review of phage family classification in different scenarios and a practical guidance for choosing appropriate taxonomic classification pipelines. To our best knowledge, this is the first review conducted under the new ICTV classification framework. The results show that the new family classification framework overall leads to better conserved groups and thus makes family-level classification more feasible.  

Contact: yannisun $@$ cityu.edu.hk  

Keywords Taxonomic classification tools $\cdot \cdot$ Bacteriophages $\cdot \cdot$ Caudoviricetes $\cdot \cdot$ Viral metagenomic data $\cdot \cdot$ Review of tools  

# 1 Introduction  

Bacteriophages (aka phages) are viruses that infect bacteria [1]. Phages are the most abundant biological entities on Earth. It is estimated that there are more than $1 0 ^ { 3 1 }$ bacteriophages on the planet, outnumbering every other organism on Earth combined [2, 3]. In most microbial communities, phages play a crucial role by shaping and maintaining microbial ecology [4, 5], facilitating co-evolutionary relationships [6, 7, 8], and promoting microbial evolution through horizontal gene transfer [9, 10].  

Phages are diverse in size, morphology, and genomic organization [11, 12]. They have a variety of structural morphologies, among which tailed double-stranded DNA (dsDNA) phages [13, 14] are the most abundant. Besides dsDNA phages, there are also phages with single-stranded DNA (ssDNA) [15], single-stranded RNA (ssRNA) [16] or double-stranded RNA (dsRNA) [17]. Phages also have a wide range of genome sizes. Recently, an increasing number of megaphages $( { > } 2 0 0 \mathrm { k b p } )$ have been sequenced, demonstrating unique genomic features [18]. Because of the high diversity of genomes, phages infecting different hosts typically have a low similarity. However, phages that infect the same host may also have considerable differences in their genomes [19, 20].  

It is now demonstrated that phages can be found in a wide variety of environments, including aquatic ecosystems [21, 22], human gut [23, 24], and soil [25, 11]. The first viral metagenome of uncultured marine viral communities was published in 2002 [26]. Phages can shape the composition and function of underlying ecosystems through two different lifestyles: temperate and virulent. Temperate phages will integrate their genomes into bacterial chromosomes and replicate with their host. They will maintain this state, which is also called prophages, until being induced by the environment’s condition, such as appropriate temperature and pH value. Then, temperate phages will enter the lytic cycle to kill the host [27, 28]. In contrast, virulent phages do not integrate their genomes into the hosts. They stay in the lytic cycle and kill the hosts after replicating themselves [29].  

The unique properties and life styles make phages key players in multiple applications. For example, phage therapy is a promising strategy for treating bacterial infections, particularly those with antibiotic-resistant bacteria. It has been found that intravenous phage preparations could treat Staphylococcus aureus that induced pneumonia in mice [30, 31]. In addition, phages can be used to treat gastrointestinal infections. It has been demonstrated that phages are effective in reducing intestinal pathogens and have less impact on the composition of the intestinal microbiota compared to antibiotics [32, 33, 34, 35]. Moreover, phages are important in food safety. The use of specific phage treatments in the food industry can prevent product spoilage and limit the spread of bacteria, providing a safe environment for animal and plant food production [36, 37, 38, 39].  

However, despite the abundance and importance of phages in various ecosystems, our understanding of phages is still very limited. According to the database supported by the National Center for Biotechnology Information (NCBI), the number of identified phages in class Caudoviricetes changed from 1,359 in 2015 to 4,483 in 2022 in the RefSeq database, which is tripled in size. Besides the reference genomes, there are roughly 63,588 assembled phages belonging to Class Caudoviricetes in the Genbank database in 2022, an almost five fold increase compared to 2015 (16,232). However, the characterization of phages cannot keep pace with the fast increase of the sequencing data.  

Assigning phages into different taxonomic groups is a fundamental step following phage discovery. The official taxonomy was established by the International Committee on Taxonomy of Viruses (ICTV) [40], which organizes viruses in several taxonomic levels, including class, order, family, subfamily, genus and so on. Within the ICTV, the Bacterial and Archaeal Viruses Subcommittee (BAVS) is responsible for the phages’ taxa. BAVS classifies phages based on a variety of phage properties, including the molecular composition of the genome (ss/ds, DNA, or RNA), the morphology, the structure of the capsid, and the host range [41]. Recently, with the increasing availability of viral genomes, using genomes for taxonomic classification has become more widely accepted [42]. Due to the extensive sequencing efforts for virus discovery, ICTV cannot catch up with the sheer number of newly identified phages, and thus many viruses are still not classified. One challenge behind this delay is the lack of standard, accurate, and comprehensive taxonomic classification tools for phages. Indeed, phage classification is not a trivial problem. The taxonomic standard in ICTV is constantly changing as new phages are discovered. Recently, ICTV updated the phage classification system in August 2022, in which several major families in the previous ICTV system are removed, such as Siphoviridae, Podoviridae, and Myoviridae. These changes can significantly affect the performance of family classification. To our best knowledge, no quantitative evaluations of the performance change have been conducted. Table 1 shows the average similarity (calculated by Dashing [43]) of the largest four families in the old and new ICTV taxonomy classification systems. The updated families are more conserved as shown by the increased average similarity, making family-level classification more feasible.  

Available taxonomic classification tools often have different designs and were tested on different datasets by their authors. Without a comprehensive comparison on the same training/reference data set and test set, it is difficult for users to choose the most appropriate solution for their needs. This paper presents a comprehensive benchmark of the main players in phage taxonomic classification under the latest ICTV standard. The remaining of this review is organized as follows. First, we will describe the main methods/models for existing phage taxonomic classification approaches and discuss whether they can be retrained/used under the new ICTV taxonomy standard. Then, we evaluate the four representative approaches that can be retrained by newly labeled sequences in different usage scenarios. In particular, we tested these tools on complete virus genomes, short contigs, simulated metagenomic datasets, and low-similarity datasets. In addition, we conducted a leave-one-family-out experiment to test whether these tools can recognize out-of-distribution sequences. By comparing their performance and analyzing the underlying reasons, we draw conclusions and provide guidance for users about choosing the most appropriate tools for different scenarios.  

Table 1: The average pairwise Dashing similarity of the four largest phage families under Caudoviricetes   


<html><body><table><tr><td colspan="2">Old version ICTV</td><td colspan="2">New version ICTV</td></tr><tr><td>Phage Family</td><td>Similarity</td><td>Phage Family</td><td>Similarity</td></tr><tr><td>Siphoviridae</td><td>0.0129</td><td>Autographiviridae</td><td>0.0171</td></tr><tr><td>Myoviridae</td><td>0.0157</td><td>Straboviridae</td><td>0.0748</td></tr><tr><td>Autographiviridae</td><td>0.0171</td><td>Herelleviridae</td><td>0.0519</td></tr><tr><td>Podoviridae</td><td>0.0206</td><td>Drexlerviridae</td><td>0.0432</td></tr></table></body></html>  

# 2 Approaches for phage taxonomic classification  

Most phage taxonomic classification approaches can output classification results in different ranks, such as order, family, and genus. In this review, we focus on comparing different tools’ performance at the family level because of the following reasons. First, the taxonomy by ICTV is under constant changes, which affects the total genus number significantly. For example, there are 735 genera in the ICTV database released in 2016. However, the number of genera increased to 2,224 in 2020. The overhaul of the genus-level taxonomy can make the definition of “ground truth” ambiguous. In addition, hundreds of rare genera only contain one phage, making the construction of reference and test set difficult. Second, classification at higher taxonomic ranks is usually easier than at lower ranks due to the smaller inter-class similarities and more abundant sequences in each class. Thus classification at orders or above is not as challenging as family classification. Caudoviricetes, a class of phage known as the tailed phages whose hosts are phage and archaea, contains the majority of the total phage sequences and can be classified by almost all of the tools mentioned above, we thus focus on the classification of the families under Caudoviricetes in this work.  

The phage taxonomic classification methods are summarized in Table 2 following the chronological order, which includes a brief description, publication year, required input data type, and the lowest predicted level of each tool. A majority of these tools conduct phage taxonomic classification based on sequence comparison, utilizing nucleotide-level or protein-level similarity between a query virus and the reference database. The comparison-based methods differ in their constructed reference database, the alignment method, and how they utilize these alignments. Both pairwise sequence alignment and hidden Markov model (HMM)-based profile alignments are commonly used. Multiple tools construct virus protein families and use them as marker genes. Using markers usually incurs less memory usage than using all phage genomes. But newly sequenced phages with novel genes may not be aligned to any marker gene families and thus cannot be assigned to a known class. Learning-based models have also been applied to phage classification. Learning models can automatically infer the sequence patterns in phage genomes of different families and use the learned features for automatic classification. A more detailed description of these tools is provided below.  

Phage Proteomic Tree [44, 56] is a relatively early program providing phage genome classification down to the family level. It extracts protein sequences from virus genomes and clusters these sequences using BLASTP [57]. Then the clusters in Phage Proteomic Tree are refined and scored. Finally, the alignment scores are converted to distances, which were used to generate the final tree using the neighbor-joining algorithm.  

Taxon-specific signature genes can be identified in most virus taxa. POGs (Phage Orthologous Groups) [45] is a collection of clusters of orthologous genes from phages, presented as profiles (multiple sequence alignment). The viral families of POGs are filtered as ‘Viruses[Organism] NOT cellular organisms [ORGN] NOT srcdb_refseq[PROP] AND vhost bacteria[filter] AND “complete genome”[All Fields] in NCBI. Signatures are extracted for each taxon, and we can use BLASTP to search for matches among the viral protein sequences. POGs are designed to be well suited for defining taxon-specific signature genes, and the profiles built from POGs are more sensitive and specific to search for signature genes in a given dataset.  

Table 2: Overview of bioinformatic approaches used for phage taxonomic classification.   


<html><body><table><tr><td>Name</td><td></td><td>Year</td><td>Description</td><td>Input data</td><td>Lowest level</td></tr><tr><td>Phage teomic [44]</td><td>Pro- Tree</td><td>2002</td><td>It uses the BLASTP distance and protein distance scores (similarity between two proteins) to generate phage proteomic trees,which can describe the relationships between different phages and can serve as a genome-based classification system for phages.</td><td>protein se- quences</td><td>Family</td></tr><tr><td>POGs [45]</td><td></td><td>2013</td><td>It provides a collection of orthologous genes clusters from phages,represented as profiles. It extracts virus-specific genes,and then classifies phages by aligning query sequences against the marker genes utilizing BLAST.</td><td>genome se- quences</td><td>Genus</td></tr><tr><td>GRAViTy [46]</td><td></td><td>2018</td><td>It conducts taxonomic classification by computing sequence relatedness between viruses using Composite Generalized Jaccard (CGJ） distances that integrate homology detection outputs and shared genomic features.</td><td>genome se- quences</td><td>Genus</td></tr><tr><td>CCP77 [47]</td><td></td><td>2019</td><td>A Phylogeny-based taxonomic classification for Caudovi- rales,inferring a concatenated Caudovirales protein (CCP77) tree based on the concatenation of protein markers using a maximum-likelihood method.</td><td>genome se- quences</td><td>Genus</td></tr><tr><td>ClassiPhage [48,49]</td><td></td><td>2019</td><td>It uses a set of phage-specific Hidden Markov Models (HMMs) generated from clusters of related proteins for phage taxonomic classification. Classiphage 2.O adds an Artificial Neutral Network(ANN) in the models.</td><td>protein se- quences</td><td>Family</td></tr><tr><td>vConTACT [50,51]</td><td></td><td>2019</td><td>A network-based application utilizing whole genome gene- sharing profiles,which integrates distance-based hierarchical clustering and confidence scores for virus classification.</td><td>protein se- quences</td><td>Genus</td></tr><tr><td>CAT [52]</td><td></td><td>2019</td><td>It can provide taxonomic classification for contigs or contig bins utilizing DIAMOND BLASTP homology search for open reading frames (ORFs).</td><td>genome se- quences</td><td>Species</td></tr><tr><td>MMseqs2 [53]</td><td></td><td>2021</td><td>A protein-search-based taxonomy classification tool for con- tigs. It assigns taxa for each possible protein product and uses weighted voting to assign taxonomic labels.</td><td>genome se- quences</td><td>Species</td></tr><tr><td>VPF-Class [54]</td><td></td><td>2021</td><td>It automates the classification by assigning the proteins to a set of Viral Protein Families(VPFs),which are then used to estimate the similarity between query genomes with classified genomes.</td><td>genome se- quences</td><td>Genus</td></tr><tr><td>PhaGCN [55]</td><td></td><td>2021</td><td>A semi-supervised learning model. It formulates the taxo- nomic classification problem as a node classification problem in a knowledge network, which is constructed by combining</td><td>genome se- quences</td><td>Family</td></tr></table></body></html>  

GRAViTy [46] also extracts protein sequences from virus genomes and cluster these sequences using BLASTP [57]. GRAViTy generates protein profile hidden Markov models (PPHMMs) and genomic organization models (GOMs) based on the sequences from BLASTP-based clustering. Then it computes Composite Generalized Jaccard (CGJ) similarity scores (a geometric mean of the two generalized Jaccard scores computed for a pair of PPHMM signatures and a pair of GOM signatures) between each sequence pair to construct the heat map and dendrogram and estimate sequences’ relatedness. GRAViTy requires users to choose reference database freely but need sequences in GenBank format as input.  

CCP77 [47] applies a concatenated protein phylogeny for the classification of tailed dsDNA viruses belonging to the specific order Caudovirales. Classiphage [48, 49] uses phage-specific Hidden Markov Models (HMMs) [58] profiles generated from clusters of related proteins for classification. The HMM profiles are built using the produced multi-sequence alignment files by the “hmmbuild” command. Classiphage 2.0 additionally trains an Artificial Neutral Network (ANN) using phage family-proteome to phage-derived HMMs scoring matrix, which can classify more phage families and include more features than its previous version.  

vConTACT [50, 51] is a high-throughput network-based approach utilizing whole-genome gene-sharing profiles. It clusters the input viral genomes together with characterized genomes. The genomes in the same cluster indicate the same family or genus, and the predicted family can be inferred if there are characterized genomes in the same cluster.  

CAT [52] provides taxonomic classification using homology searches. It uses DIAMOND BLASTP to identify homologous sequences and then assigns query sequences into taxa with a voting approach. The authors of CAT show that using the best hit strategy can lead to low specificity and thus design a more robust strategy based on multiple hits. Users can select the reference database and tune the setting, which is more flexible than some other tools. Moreover, it has a very low memory usage.  

MMseqs2 [53] is a fast contig taxonomic assignment tool. Similar to CAT, it conducts protein homology search against reference sequences and uses majority vote to assign the most specific taxon for a contig. With some optimizations and adoption of 2bLCA [59], MMseqs2 circumvents the need of adjusting a parameter in CAT and achieves faster speed on the tested bacterial and eukaryotic datasets. It allows users to supply a customized reference database.  

VPF-Class [54] provides both taxonomic classification and host prediction for input viral genomes. It compares predicted proteins against the set of constructed Viral Protein Families (VPFs) (from the IMG/VR system). Then it derives taxonomic classifications and confidence scores from the list of VPFs detected on each query genome. However, VPF-Class does not require users to download and select the reference datasets.  

PhaGCN [55] is a semi-supervised learning model for phage taxonomic classification developed by our team. This model constructs a knowledge graph by combining the DNA sequence features learned by Convolutional Neural Networks (CNN) and protein sequence similarity gained from the gene-sharing network. The learning model can incorporate the automatically learned features for each contig. However, unlike sequence comparison-based approaches, PhaGCN only accepts phage-like sequences as input. Thus, a pre-processing step is needed for detecting those contigs from metagenomic data. A number of tools, such as VirFinder [60], Seeker [61], and PhaMer [62] can be applied in the pre-processing step.  

# 3 Experiments and Results  

Because of the changes in the ICTV classification system, the models/reference databases need to be updated using the latest labeled sequences. However, not all the tools in Table 2 can be updated easily. Among them, only CAT, GRAViTy, PhaGCN, MMseqs2, and vConTACT 2.0 allow users to change their reference databases or retrain the models with reasonable efforts. The others do not specify the feasibility of changing models or reference databases in the descriptions. The source code of CCP77 is only available on request but not to the public. The code of GRAViTy released at GitHub is the alpha version and the author mentioned that they are currently working on a new and improved version that is more user-friendly and written in python3. Nevertheless, we downloaded and installed the alpha version of GRAViTy. The alpha version is computationally expensive and requires 30 hours to build a reference database with about 1200 genomes and another 25 hours to process just 300 queries. Therefore, we focus on evaluating the performance of the four tools: PhaGCN, vConTACT 2.0, CAT, and MMseqs2. These tools were recently published and demonstrated good performance in their own or others’ tests. In addition, the corresponding codes and tools are still under maintenance. None of them requires an internet connection or a web server. To mimic the scenario of applying these tools to datasets without known taxonomic composition, we apply all these tools with their default parameters, which are optimized by the authors. The commands for running all these tools are available in the Supplementary File. All the tools were run on Intel $\mathrm { V R } ^ { \mathbb { G } } \mathrm { X e o n V R } ^ { \mathbb { G } }$ Gold 6258 R CPU with 8 cores.  

Table 3: The 19 families under Class Caudoviricetes from the RefSeq database we used in the experiments. Number the number of complete sequences in each family.   


<html><body><table><tr><td>Family Name</td><td>Number</td><td>Family Name</td><td>Number</td></tr><tr><td>Autographiviridae</td><td>370</td><td>Straboviridae</td><td>204</td></tr><tr><td>Herelleviridae</td><td>127</td><td>Drexlerviridae</td><td>117</td></tr><tr><td>Demerecviridae</td><td>94</td><td>Peduoviridae</td><td>83</td></tr><tr><td>Casjensviridae</td><td>76</td><td>Schitoviridae</td><td>76</td></tr><tr><td>Kyanoviridae</td><td>62</td><td>Ackermannviridae</td><td>62</td></tr><tr><td>Rountreeviridae</td><td>35</td><td>Salasmaviridae</td><td>34</td></tr><tr><td>Vilmaviridae</td><td>31</td><td>Zierdtviridae</td><td>26</td></tr><tr><td>Mesyanzhinovviridae</td><td>17</td><td>Chaseviridae</td><td>14</td></tr><tr><td>Zobellviridae</td><td>13</td><td>Orlajensenviridae</td><td>11</td></tr><tr><td>Guelinviridae</td><td>8</td><td>Total</td><td>1460</td></tr></table></body></html>  

# 3.1 Dataset  

We rigorously evaluated these phages taxonomic classification tools on multiple datasets. The detailed information is sted below.  

The RefSeq dataset RefSeq is a widely used benchmark dataset in phage classification tasks. By October 2022, there are 1,826 complete sequences with family-label under Class Caudoviricetes in the RefSeq database. In this paper, we only focus on the phages infecting bacteria. After filtering out the families that infect archaeas or contain sequences less than 6, there are 19 families (including 1460 complete sequences) we can use in our experiments. Table 3 shows the number of sequences within the 19 families under class Caudoviricetes, among which Autographiviridae contains the largest number of sequences. For the tools that require protein sequences, we used Prodigal [63] to predict and translate the nucleotide sequence into the proteins. We sorted the sequence by their release time at RefSeq. Then, we used the first $80 \%$ of the labeled complete sequences from each family as the training set/reference database to retrain/update the four tools, and the rest $20 \%$ as test set. Because we split the data in chronological order, the data in the test set are more recent (almost all were released in 2020 or after).   
Short contigs dataset This dataset contains segments with different lengths, including 500 bp, 1,000 bp, 3,000 bp, 5,000 bp, 10,000 bp, and 15,000 bp. We randomly generated the segments from the $20 \%$ RefSeq dataset (293 sequences) mentioned above. For each length, we cut ten segments from each phage genome by selecting a random start position. Finally, we had 2, 930 phage contigs for each length and 29,300 for all different lengths. Then, we used these segments to evaluate the performance of the four tools on short contigs. Simulated metagenomic dataset We used a simulated metagenomic dataset generated by six common bacteria living in human gut [62]. We first utilized metaSPAdes [64] to assemble the reads into contigs. Then PhaMer [62] was applied to identify bacteriophages from metagenomic data, and the labels of the contigs were determined using BLAST [65]. Eventually, 37 contigs were used in the experiments. More details about this dataset will be provided in the section of Experiment 4.   
Low-similarity dataset To test the tools’ performance on classifying highly diverged phages, we constructed a hard case where the test sequences share low similarity with the reference database/training data. Specifically, we calculated the Dashing pairwise similarity of the sequences in each family and then used the approach in [66] to partition the data into two parts with specified maximum similarity. With this method, we got 264 and 45 genomes for training and test, where each test genome has at most 0.015 Dashing similarity with any reference genome. Then we randomly cut 15 contigs with a length of 3,000 bp and 5,000 bp, respectively, from each testing genome. Finally, there are 675 contigs for each length in the test set.  

# 3.2 Evaluating criteria for different tools  

# 3.2.1 Metrics  

An ideal phage classification tool should assign correct labels for as many inputs as possible. Nevertheless, there is usually a tradeoff between the percentage of prediction and the accuracy of the prediction. Some tools may sacrifice the percentage of prediction in order to achieve high specificity and accuracy, while others may predict more with lower accuracy. Thus the first metric is prediction rate, which is the ratio of outputs with prediction results $( N _ { p r e d }$ in Equation 1) to the total input $ { \mathrm { ~ \textrm ~ { ~ ~ } ~ } } _ { N _ { a l l } }$ in Equation 1). Because some tools only provide a family name as output, commonly used metrics such as AUROC cannot be computed. In this work, we calculated accuracy, recall, and precision for each tool (Equations 2-4). $N _ { c o r r e c t }$ is the number of sequences with correct predictions in output. $N _ { t o t a l }$ is the total number of sequences used to evaluate, which can be $N _ { a l l }$ or $N _ { p r e d }$ when we report accuracy for all input phage sequences $( N _ { a l l } )$ or only for sequences with predictions in output $( N _ { p r e d } )$ , respectively. Providing accuracy for all input sequences has the advantage of using the same denominator (i.e. $N _ { a l l }$ ) for all tools. But it penalizes the tools of low prediction rate twice. On the other hand, reporting accuracy for only sequences with predictions removes the impact of prediction rate but may favor tools with low prediction rate (i.e. small $N _ { p r e d , }$ ). Thus, reporting both can provide a more comprehensive evaluation for users. For example, if there are 293 $( N _ { a l l } )$ sequences input, among which 290 sequences have classification prediction results $( N _ { p r e d } )$ , and 285 of them have correct results $( N _ { c o r r e c t } )$ , the accuracy on all input will be $2 8 5 / 2 9 3 { = } 0 . 9 7 3$ , and the accuracy on predicted sequences will be $2 8 5 / 2 9 0 { = } 0 . 9 8 3$ . We only calculate the recall and precision of each family $( P r e c i s i o n _ { i }$ and $R e c a l l _ { i }$ ) to check the performance on different families. $T P _ { i }$ , $F P _ { i }$ , and $F N _ { i }$ are the true positive, false positive, and false negative for family $i$ , respectively.  

$$
P r e d i c t i o n r a t e = \frac { N _ { p r e d } } { N _ { a l l } }
$$  

$$
A c c u r a c y = \frac { N _ { c o r r e c t } } { N _ { t o t a l } }
$$  

$$
P r e c i s i o n _ { i } = \frac { T P _ { i } } { T P _ { i } + F P _ { i } }
$$  

$$
R e c a l l _ { i } = \frac { T P _ { i } } { T P _ { i } + F N _ { i } }
$$  

# 3.2.2 Description of the output  

Because the output format of each tool is different, we will describe how we process the output and calculate the metrics in detail.  

vConTACT 2.0 can output the result of each sequence and assign it a “VC State”, including “Singleton”, ‘Outlier”, or “Clustered”. In addition, the sequences with a “Clustered” state will be assigned to a VC cluster/subcluster. When the query sequence is within the same VC cluster as a reference genome, the taxonomic labels can be assigned based on the known labels. However, some sequences are clustered but have no reference genome in the same VC cluster, so they can not be assigned with a known label. Therefore, we treat the sequence with VC state of “Singleton”, “Outlier”, and “Clustered” but no reference genome in the same clusters, as “no prediction”. In other words, $N _ { p r e d }$ of vConTACT 2.0 refers to the number of the sequences that are clustered with reference genomes.  

PhaGCN will not output the classification results for the sequences they can not classify, so $N _ { p r e d }$ of PhaGCN is the number of sequences that can be predicted.  

MMseqs2 and CAT will not output any prediction result for the sequences they cannot classify. The classification result of MMseqs2 and CAT can be a label at different ranks. If the prediction at the lowest rank is above family, we also treat this sequence as “no prediction” for the family level. The number of the rest sequences is $N _ { p r e d }$ of MMseqs2/CAT.  

# 3.3 Experiment 1: Leave-one-family-out experiments  

The constant change of ICTV underscores a need for classification tools to recognize the sequences that are not part of the current classification system. For example, the three largest families, Siphoviridae, Podoviridae, and Myoviridae, were largely removed from the current ICTV system. Some of the sequences that belonged to these three families are not part of any existing family. Thus, the classification tools need to handle these out-of-distribution sequences by providing a signal for users.  

To examine whether the tested tools can single out those out-of-distribution sequences, we removed all the phages in one family from the training data and retrained the models. Then the retrained models are applied to the removed family members. Ideally, the test sequences in this removed family should not be classified into any existing family labels.  

At first, we conducted the experiments on a small and a relatively large family: Guelinviridae and Rountreeviridae. The classification results are plotted in Fig. 1 and Fig. 2, which show that PhaGCN assigned all of the query genomes to one of the other families in the training set, while CAT and MMseqs2 can correctly recognize a few sequences as “no family label”. However, vConTACT 2.0 can assign all sequences to “Outlier/Singleton” or a “VC cluster” without reference genomes.  

![](images/be6523f52e85d02368929fa25883b2809ab8632fb11c4653371b632de431e1a7.jpg)  
Figure 1: The classification result of Guelinviridae sequences in tools that are retrained by removing all Guelinviridae sequences. “independent clustered”: the sequences are in a VC cluster without any reference genome.  

![](images/18351918d88fe5a87aa47ff08aa8ecd359093a7e98c02d6fc6c61da1484ac895.jpg)  
Figure 2: The classification result of Rountreeviridae sequences in tools that are retrained by removing all Rountreeviridae sequences. “independent clustered”: the sequences are in a VC cluster without any reference genome.  

We then extended the experiment to each family. Because the current version of PhaGCN is not designed to handle out-of-distribution sequences, we only show the results for CAT, MMseqs2, and vConTACT 2.0 in Table 4. The output of these three tools for the test sequences are divided into two parts: those that did not output a family label (“no prediction”, defined in the section Description of the output), and those that can output a family label from the training data (i.e., a misclassification in this experiment). Table 4 shows the misclassification rate of each tool. CAT and MMseqs2 assign more test sequences to other families in the reference database. In contrast, vConTACT 2.0 can assign almost all sequences of each family to “Outlier/Singleton” labels or “VC cluster” without reference genomes. The misclassification rates of CAT and MMseqs2 vary widely across different families, with the ranges 0-1 and 0-0.92, respectively. A closer look at those results reveals that the misclassified phages tend to distribute in a small set of families. For example, almost all sequences belonging to Guelinviridae are classified into Salasmaviridae by CAT, which is likely due to the higher inter-family similarity between them. Specifically, $2 9 . 6 \%$ proteins of Guelinviridae can align with Salasmaviridae using BLASTP. Similarly, sequences from Zobellviridae tend to be classified into family Autographiviridae because they share about $1 6 . 9 \%$ proteins. Therefore, the inter-family similarity is an essential factor leading to misclassification. Overall, the misclassification results of MMseqs2 are more divergent than CAT. For example, CAT will classify Autographiviridae genomes into 4 other families, while MMseqs2 will assign them into 8 families (including the 4 families in CAT).  

Then we extended the experiment to the genomes that are unclassified at the family level in the RefSeq database under class Caudoviricetes. Because the three largest families Myoviridae, Siphoviridae and Podoviridae were removed, we used the genome sequences that initially belonged to these three families but now no longer have a family label as the test data. There are 2445 of them, and the classification result is shown in Fig. 3. MMseqs2 and CAT misclassified about $65 \%$ of the input sequences. vConTACT 2.0 can identify $98 \%$ unclassified sequences by assigning them in independent clusters or outputting a “Singleton/Outlier” label and only misclassified $2 \%$ sequences. In conclusion, vConTACT 2.0 performs better in identifying novel phages than the other three tools.  

Table 4: The percentage of misclassified sequences in leave-one-family-out experiment for each family.   


<html><body><table><tr><td colspan="4"></td></tr><tr><td>Family Name</td><td>CAT</td><td>MMseqs2</td><td>vConTACT 2.0</td></tr><tr><td>Autographiviridae</td><td>0.78</td><td>0.58</td><td>0.12</td></tr><tr><td>Straboviridae</td><td>0.05</td><td>0</td><td>0</td></tr><tr><td>Herelleviridae</td><td>0.02</td><td>0.2</td><td>0</td></tr><tr><td>Drexlerviridae</td><td>0.3</td><td>0.25</td><td>0.01</td></tr><tr><td>Demerecviridae</td><td>0</td><td>0.32</td><td>0</td></tr><tr><td>Peduoviridae</td><td>0.55</td><td>0.84</td><td>0</td></tr><tr><td>Casjensviridae</td><td>0.86</td><td>0.87</td><td>0</td></tr><tr><td>Schitoviridae</td><td>0.24</td><td>0.21</td><td>0</td></tr><tr><td>Kyanoviridae</td><td>0</td><td>0.18</td><td>0</td></tr><tr><td>Ackermannviridae</td><td>0</td><td>0.02</td><td>0</td></tr><tr><td>Rountreeviridae</td><td>0.86</td><td>0.91</td><td>0</td></tr><tr><td>Salasmaviridae</td><td>0.88</td><td>0.82</td><td>0.44</td></tr><tr><td>Vilmaviridae</td><td>0.13</td><td>0.45</td><td>0</td></tr><tr><td>Zierdtviridae</td><td>0.81</td><td>0.92</td><td>0</td></tr><tr><td>Mesyanzhinovviridae</td><td>0</td><td>0.06</td><td>0</td></tr><tr><td>Chaseviridae</td><td>0</td><td>0.14</td><td>0</td></tr><tr><td>Zobellviridae</td><td>0.77</td><td>0.38</td><td>0</td></tr><tr><td>Orlajensenviridae</td><td>0.73</td><td>0.18</td><td>0</td></tr><tr><td>Guelinviridae</td><td>1.0</td><td>0.88</td><td>0</td></tr><tr><td>Average</td><td>0.42</td><td>0.43</td><td>0.03</td></tr><tr><td>MMseqs2 66% 34% misclassified</td><td>CAT 62% 38%</td><td>93%</td><td>vConTACT2.0 2%</td></tr></table></body></html>  

# 3.4 Experiment 2: classification performance  

As we described in Section “Dataset”, we used $20 \%$ (293) of the complete sequences from the RefSeq database as the test set, and the other $80 \%$ as the reference/training set. To mimic metagenomic assembled contigs, we generated six sets of segments of different lengths for comparison, including 500 bp, 1,000 bp, 3,000 bp, 5,000 bp, 10,000 bp, and 15,000 bp. We randomly selected the start positions for each length and cut ten segments from each complete sequence. Finally, we had 2,930 phage fragments for each length and 29,593 for all different lengths as the test data (293 complete sequences $+ 2 9 3 0 * 1 0$ short fragments).  

A good taxonomic classification tool should have a high prediction rate and high accuracy. First, we recorded the prediction rate of each tool on different lengths. Because PhaGCN only accepts contigs longer than 2,000 bp, we do not show its results on 500 bp and 1,000 bp in Fig. 4. The prediction rate (Fig. 4 (A)) of all tools becomes higher with the increase in sequence length. This is expected because longer sequences usually provide more information for classification. Almost all pipelines can maintain a high prediction rate $( > 8 0 \% )$ on short sequences except vConTACT 2.0. PhaGCN has the highest prediction rate if the inputs are longer than 5,000 bp, while CAT is slightly lower. vConTACT 2.0 is mainly designed for complete or long sequences, and its prediction rate drops sharply when the inputs are shorter than 15,000 bp. All four can handle more than $9 5 \%$ of complete sequences, among which PhaGCN can predict all of them $( 1 0 0 \% )$ , and the prediction rates of MMseqs2, CAT, and vConTACT 2.0 are $9 9 . 3 \%$ , $9 7 . 9 \%$ , and $9 5 . 1 \%$ , respectively.  

![](images/eaf15d0e5fcdc04abcc206e0e0f47989d5f29c20af31d92286c222408a837056.jpg)  
Figure 4: The performance of each tool on contigs from the RefSeq. (A): the prediction rate of four tools on different lengths. (B): the accuracy of four tools on phage contigs with predictions. (C): the accuracy of four tools on all input phage contigs. X-axis: the lengths Y-axis: the values.  

Fig. 4 (B) shows the accuracy of the four tools on phage sequences with predictions $N _ { p r e d }$ in Equation 1). Similar to the prediction rates above, the accuracy of these approaches becomes better as the sequence lengths increase. The classification ability of CAT, PhaGCN, and MMseqs2 are not significantly affected by the change of contig lengths. On incomplete contigs, the accuracy of vConTACT 2.0 has an obvious upward trend when length increases. CAT gains the best prediction accuracy for contigs longer than 5,000 bp. Combined with the slightly lower prediction rate of CAT mentioned above, we can conclude that there is a tradeoff between the prediction rate and the accuracy of CAT. The accuracy of PhaGCN is slightly lower than the other two on contigs, and all three tools reach a high accuracy $( 1 0 0 \% )$ for all complete sequences with predictions.  

Fig. 4 (C) shows the accuracy of the four tools on all input phage contigs $ { \mathrm { ~ \textrm ~ { ~ ~ } ~ } } N _ { a l l }$ in Equation 1), which combines the results in (A) and (B) in order to display the overall performance of each tool. It reveals that PhaGCN keeps the best performance on contigs longer than 5,000 bp and reaches $100 \%$ accuracy on complete genomes because it gains $100 \%$ accuracy and prediction rate in (A) and (B), respectively. It is worth noting that the other three tools all have a less than $100 \%$ recall on Autographiviridae, most likely due to the lower pairwise similarity in Autographiviridae (Table 1). Due to the length limitation of PhaGCN, it is not suitable for classifying contigs shorter than 2,000 bp. When classifying contigs longer than 2,000 bp, PhaGCN and MMseqs2 are recommended for obtaining high prediction rates. Otherwise, CAT is a better choice if precision is the primary consideration.  

![](images/02b7f6002d3e31334a3ab65214b33f96201230281c5052e9c27c38874aa324e5.jpg)  
Figure 5: (A) The prediction rate of four tools with reduced reference datasets. (B) The corresponding accuracy on sequences with predictions. $\mathbf { \boldsymbol { X } }$ -axis: the tools and training data partitions Y-axis: the values.  

![](images/03881040fe71a8eb8f36b97d2536eca7ed38e3aa1fc703681252d76406708218.jpg)  

# 3.5 Experiment 3: impact of training set size on classification performance  

Being a learning-based classification tool, PhaGCN can be affected by training data size. To test whether PhaGCN and other alignment-based tools suffer from reduced training data/reference database, we used $80 \%$ (the same as Experiment 2), $60 \%$ , and $50 \%$ of the RefSeq databases as the reference database for these tools, respectively. Then we tested them on the same test set as in Experiment 2. As shown in Fig. 5 (A), the prediction rates of PhaGCN with different reference databases have no obvious differences. There is a slight change in the prediction rate of CAT, MMseqs2, and vConTACT 2.0, but the differences do not exceed $0 . 2 \%$ . In addition, the accuracy of these tools shown in Fig. 5 (B) are almost identical and are less affected than the prediction rate.  

![](images/22ceae3d0cc531e96253ea455d905ed0d3e3536c44ff5c6594dca2e7b33a7052.jpg)  
Figure 6: (A) The performance of the four tools on the simulated metagenomic dataset. The bars show the accuracy on all inputs. The top part with patterns in vConTACT 2.0 shows the percentage of contigs that are not clustered with any reference genome. (B) The performance of each tool on the two low-similarity datasets. Each bar shows the tools’ accuracy on all input contigs.  

# 3.6 Experiment 4: classification performance on the simulated metagenomic dataset  

In this experiment, we used the simulated metagenomic dataset provided in PhaMer [62]. The dataset is a small-scale metagenomic dataset simulated by CAMISIM [67] using the commonly seen bacteria living in the human gut and the phages that infect these bacteria. The reads were assembled into contigs using metaSPAdes [64].  

Table 5: Family composition of the simulated metagenomic dataset.   


<html><body><table><tr><td>Family Name</td><td>Number</td></tr><tr><td>Straboviridae</td><td>28</td></tr><tr><td>Drexlerviridae</td><td>6</td></tr><tr><td>Demerecviridae</td><td>1</td></tr><tr><td>Peduoviridae</td><td>1</td></tr><tr><td>Ackermannviridae</td><td>1</td></tr><tr><td>Total</td><td>37</td></tr></table></body></html>  

We kept contigs of size above 3,000 bp. To assign labels to the contigs, we used BLAST [65] to map contigs to reference genomes and calculated the coverage. Only the contigs with at least $90 \%$ of the sequence aligning to a reference genome were kept. Others are likely chimeric contigs due to assembly errors and thus are not used for testing. Finally, the number of contigs we could use in the experiment is 37. The name of the families and the number of genomes within each family are listed in Table 5. Compared to Table 3, this test set contains a different abundance distribution for the component families, which can thus change the performance of these tools.  

As shown in Fig. 6 (A), PhaGCN, MMseqs2, and CAT can classify all the simulated sequences correctly, which is slightly higher than that on the RefSeq data in Experiment 2. A plausible reason is that most of the sequences in this simulated dataset belong to Straboviridae and Ackermannviridae, which make up a large part of the reference database according to Table 3 ( $14 \%$ and $4 \%$ ). In addition, they have greater intra-family similarities. The performance of vConTACT 2.0 is lower than the other three tools because the assembled contigs are short. This experiment shows that PhaGCN, MMSeq2, and CAT can process assembled contigs with different lengths.  

# 3.7 Experiment 5: classification performance on the low-similarity dataset  

Although the updated families under the new ICTV standard exhibit higher pairwise sequence similarity, there are still some diverged members. The diverged members may appear more often when sequencing new or underrepresented ecosystems. Thus, we test these tools’ performance on predicting highly diverged sequences using the “low similarity dataset”. There are 45 genomes in the test set with the maximum Dashing similarity of 0.015 with any reference genome. Then we randomly cut 15 contigs with a length of 3,000 bp and 5,000 bp from each test genome, leading to 1,350 contigs in total. Fig. 6 (B) shows the accuracy of all inputs. Because vConTACT 2.0 can not handle short contigs, we exclude it from this experiment.  

Fig. 6 (B) reveals that the accuracy of MMseqs2 decreases by more than $10 \%$ compared to Fig. 4 (C) from Experiment 2. And the accuracy drop in CAT $( 6 \%$ , $5 . 2 \% )$ are greater than PhaGCN $( 3 . 3 \%$ , $2 \% )$ on the contigs of the same lengths. Therefore, the increased divergence between test and training data has a greater impact on alignment-based tools than PhaGCN in this experiment.  

# 3.8 Comparison of Running Time  

Running time is also an essential factor to consider for practical usage. Table 5 shows the running time of the tools for processing 500 complete sequences in RefSeq when using a different number of CPUs. Users can save more time by increasing the number of CPUs. The table also shows that CAT and MMseqs2 take the least time to process 500 complete phages.  

Table 6: The total running time of tools for classifying 500 genomes using a different number of CPUs. All the tools are run on Intel $\mathrm { V R } ^ { \mathrm { @ } } \mathrm { X e o n V R ^ { \mathrm {  } } }$ Gold 6258 R CPU with 8 cores.   


<html><body><table><tr><td>Time (min)</td><td>PhaGCN</td><td>MMseqs2</td><td>CAT</td><td>vConTACT2.0</td></tr><tr><td>1 CPU</td><td>23</td><td>2</td><td>3</td><td>141</td></tr><tr><td>4 CPUs</td><td>18</td><td>1</td><td>2</td><td>64</td></tr></table></body></html>  

# 4 Discussion and conclusion  

This work presents a review of taxonomic classification tools on phage family classification under Caudoviricetes. To our best knowledge, this is the first review under the new ICTV standard released in August 2022. Compared to the previous version of ICTV, the updated families in the latest system are more conserved, which warrants a high prediction rate and accuracy of alignment-based tools. For example, the prediction rate of CAT and vConTACT 2.0 were $62 \%$ and $92 \%$ on the data in the previous ICTV system, respectively. And their accuracy on complete genomes were only $6 1 . 7 \%$ and $8 6 \%$ . However, their prediction rate and accuracy are significantly better under the new classification system.  

The constant change of the taxonomic classification system by ICTV emphasizes the need for a tool to provide database updating or model retraining. Tools without these utilities can return obsolete or even wrong labels, making their practical usage limited. Many of these tools in Table 2 either lack this option or need excessive efforts to retrain.  

Despite great efforts, the current classification system by ICTV is not complete. New families can appear with new viruses sequenced and discovered, particularly those from underrepresented ecosystems. Thus, it is desired that a classification tool can handle out-of-distribution inputs, which are not part of any existing families. Based on our leave-one-family-out experiment, vConTACT 2.0 is more sensitive to those out-of-distribution sequences than others. However, a price paid by vConTACT 2.0 is its low prediction rate on short contigs, which is likely caused by the low gene sharing significance score between the query and the reference. Other tools perform better on short contigs, which is important for virus composition analysis in metagenomic data.  

PhaGCN can only classify sequences on the family level. The lowest levels that the other three tools can classify are genus level or below. The experimental results show that all of them can perform well on complete genomes from the RefSeq database after retraining. PhaGCN has the highest prediction rate when classifying short contigs $( > 3 , 0 0 0$ bp), and CAT gains a higher accuracy with a slightly lower prediction rate. Therefore, when classifying incomplete contigs larger than 3,000 bp, PhaGCN, CAT, and MMseqs2 can all be considered, but PhaGCN has a better overall performance. In addition, CAT and MMseqs2 can be used to classify contigs shorter than 2,000 bp because PhaGCN can not handle that length. All these four tools are robust against the size reduction of the reference database/training data. The performance of PhaGCN is less affected in classifying highly diverged sequences that share low similarity with the reference genomes.  

The focus of this review is family-level classification. While the current families annotated by ICTV usually contain multiple phages per family, the genus size distribution exhibits a much more skewed distribution with many genera only containing one phage genome. It is not trivial to create appropriate reference database/training data vs. test data with hundreds of rare genera. It is our future work to examine the impact of the long tail distribution on current classification tools.  

# Data Availability  

The detailed information of the code and datasets is provided in the supplementary file.  

# Funding  

City University of Hong Kong (Project 9678241 and 7005453) and the Hong Kong Innovation and Technology Commission (InnoHK Project CIMDA).  

# References  

[1] Stephen McGrath, D van Sinderen, et al. Bacteriophage: genetics and molecular biology. Caister Academic Press, 2007. [2] Curtis A Suttle. Viruses in the sea. Nature, 437(7057):356–361, 2005. [3] Scott LaFee and Heather Buschman. Novel phage therapy saves patient with multidrug-resistant bacterial infection. UC San Diego News Center, University of California, USA, 2017.   
[4] Britt Koskella and Sean Meaden. Understanding bacteriophage specificity in natural microbial communities. Viruses, 5(3):806–823, 2013. [5] T Frede Thingstad. Elements of a theory for the mechanisms controlling abundance, diversity, and biogeochemical role of lytic bacterial viruses in aquatic systems. Limnology and Oceanography, 45(6):1320–1328, 2000.   
[6] Ana Georgina Cobián Güemes, Merry Youle, Vito Adrian Cantú, Ben Felts, James Nulton, and Forest Rohwer. Viruses as winners in the game of life. Annual Review of Virology, 3:197–214, 2016.   
[7] Lesley Hoyles, Anne L McCartney, Horst Neve, Glenn R Gibson, Jeremy D Sanderson, Knut J Heller, and Douwe van Sinderen. Characterization of virus-like particles associated with the human faecal and caecal microbiota. Research in Microbiology, 165(10):803–812, 2014. [8] Cynthia B Silveira and Forest L Rohwer. Piggyback-the-winner in host-associated microbial communities. NPJ Biofilms and Microbiomes, 2(1):1–5, 2016.   
[9] Maryury Brown-Jaque, William Calero-Cáceres, and Maite Muniesa. Transfer of antibiotic-resistance genes via phage-related mobile elements. Plasmid, 79:1–7, 2015.   
[10] Yin Ning Chiang, José R Penadés, and John Chen. Genetic transduction by phages and chromosomal islands: The new and noncanonical. PLoS Pathogens, 15(8):e1007878, 2019.   
[11] Cheryl-Emiliane T Chow and Curtis A Suttle. Biogeography of viruses in the sea. Annu Rev Virol, 2(1):41–66, 2015.   
[12] Hans-W Ackermann. Classification of bacteriophages. The Bacteriophages, 2:8–16, 2006.   
[13] Kathryn M Kauffman, Fatima A Hussain, Joy Yang, Philip Arevalo, Julia M Brown, William K Chang, David VanInsberghe, Joseph Elsherbini, Radhey S Sharma, Michael B Cutler, et al. A major lineage of non-tailed dsDNA viruses as unrecognized killers of marine bacteria. Nature, 554(7690):118–122, 2018.   
[14] Jennifer R Brum, Ryan O Schenck, and Matthew B Sullivan. Global morphological analysis of marine viruses shows minimal regional variation and dominance of non-tailed viruses. The ISME Journal, 7(9):1738–1751, 2013.   
[15] Efrem S Lim, Yanjiao Zhou, Guoyan Zhao, Irma K Bauer, Lindsay Droit, I Malick Ndao, Barbara B Warner, Phillip I Tarr, David Wang, and Lori R Holtz. Early life dynamics of the human gut virome and bacterial microbiome in infants. Nature Medicine, 21(10):1228–1234, 2015.   
[16] Tim Loeb and Norton D Zinder. A bacteriophage containing RNA. Proceedings of the National Academy of Sciences, 47(3):282–289, 1961.   
[17] Peter Mertens. The dsRNA viruses. Virus Research, 101(1):3–13, 2004.   
[18] Yihui Yuan and Meiying Gao. Jumbo bacteriophages: an overview. Frontiers in Microbiology, 8:403, 2017.   
[19] Graham F Hatfull. Bacteriophage genomics. Current Opinion in Microbiology, 11(5):447–453, 2008.   
[20] Mart Krupovic, David Prangishvili, Roger W Hendrix, and Dennis H Bamford. Genomics of bacterial and archaeal viruses: dynamics within the prokaryotic virosphere. Microbiology and Molecular Biology Reviews, 75(4):610–635, 2011.   
[21] John H Paul, Matthew B Sullivan, Anca M Segall, and Forest Rohwer. Marine phage genomics. Comparative Biochemistry and Physiology Part B: Biochemistry and Molecular Biology, 133(4):463–476, 2002.   
[22] Burton Guttman, Raul Raya, and Elizabeth Kutter. Basic phage biology. Bacteriophages: Biology and Applications, 4:30–63, 2005.   
[23] Thomas DS Sutton and Colin Hill. Gut bacteriophage: current understanding and challenges. Frontiers in Endocrinology, 10:784, 2019.   
[24] Pilar Manrique, Michael Dills, and Mark J Young. The human gut phage community and its implications for health and disease. Viruses, 9(6):141, 2017.   
[25] Kurt E Williamson, Jeffry J Fuhrmann, K Eric Wommack, Mark Radosevich, et al. Viruses in soil ecosystems: an unknown quantity within an unexplored territory. Annu Rev Virol, 4(1):201–219, 2017.   
[26] Mya Breitbart, Peter Salamon, Bjarne Andresen, Joseph M Mahaffy, Anca M Segall, David Mead, Farooq Azam, and Forest Rohwer. Genomic analysis of uncultured marine viral communities. Proceedings of the National Academy of Sciences, 99(22):14250–14255, 2002.   
[27] Allan Campbell. The future of bacteriophage biology. Nature Reviews Genetics, 4(6):471–477, 2003.   
[28] Cristina Howard-Varona, Katherine R Hargreaves, Stephen T Abedon, and Matthew B Sullivan. Lysogeny in nature: mechanisms, impact and ecology of temperate phages. The ISME Journal, 11(7):1511–1520, 2017.   
[29] Zack Hobbs and Stephen T Abedon. Diversity of phage infection types and associated terminology: the problem with ‘Lytic or lysogenic’. FEMS Microbiology Letters, 363(7), 2016.   
[30] Emilie Saussereau and Laurent Debarbieux. Bacteriophages in the experimental treatment of Pseudomonas aeruginosa infections in mice. Advances in Virus Research, 83:123–141, 2012.   
[31] Joseph M Ochieng’ Oduor, Nyamongo Onkoba, Fredrick Maloba, and Atunga Nyachieo. Experimental phage therapy against haematogenous multi-drug resistant staphylococcus aureus pneumonia in mice. African Journal of Laboratory Medicine, 5(1):1–7, 2016.   
[32] Matthieu Galtier, Luisa De Sordi, Damien Maura, Harindra Arachchi, Stevenn Volant, Marie-Agnès Dillies, and Laurent Debarbieux. Bacteriophages to reduce gut carriage of antibiotic resistant uropathogens with low impact on microbiota composition. Environmental Microbiology, 18(7):2237–2245, 2016.   
[33] Janet Y Nale, Janice Spencer, Katherine R Hargreaves, Anthony M Buckley, Przemysław Trzepi´nski, Gillian R Douce, and Martha RJ Clokie. Bacteriophage combinations significantly reduce Clostridium difficile growth in vitro and proliferation in vivo. Antimicrobial Agents and Chemotherapy, 60(2):968–981, 2016.   
[34] Abhishek Jaiswal, Hemanta Koley, Amit Ghosh, Anup Palit, and Banwarilal Sarkar. Efficacy of cocktail phage therapy in treating Vibrio cholerae infection in rabbit model. Microbes and Infection, 15(2):152–156, 2013.   
[35] Beatriz Gutiérrez and Pilar Domingo-Calap. Phage therapy in gastrointestinal diseases. Microorganisms, 8(9):1420, 2020.   
[36] Sanna M Sillankorva, Hugo Oliveira, and Joana Azeredo. Bacteriophages and their role in food safety. International Journal of Microbiology, 2012, 2012.   
[37] P Garcia, B Martinez, JM Obeso, and A Rodriguez. Bacteriophages and their application in food safety. Letters in Applied Microbiology, 47(6):479–485, 2008.   
[38] Brid Coffey, Susan Mills, Aidan Coffey, Olivia McAuliffe, R Paul Ross, et al. Phage and their lysins as biocontrol agents for food safety applications. Annu. Rev. Food Sci. Technol, 1:449–468, 2010.   
[39] Diana Gutiérrez, Lorena Rodríguez-Rubio, Lucía Fernández, Beatriz Martínez, Ana Rodríguez, and Pilar García. Applicability of commercial phage-based products against Listeria monocytogenes for improvement of food safety in Spanish dry-cured ham and food contact surfaces. Food Control, 73:1474–1482, 2017.   
[40] Michael J Adams, Elliot J Lefkowitz, Andrew MQ King, Balázs Harrach, Robert L Harrison, Nick J Knowles, Andrew M Kropinski, Mart Krupovic, Jens H Kuhn, Arcady R Mushegian, et al. 50 years of the International Committee on Taxonomy of Viruses: progress and prospects. Archives of Virology, 162(5):1441–1446, 2017.   
[41] Moira B Dion, Frank Oechslin, and Sylvain Moineau. Phage diversity, genomics and phylogeny. Nature Reviews Microbiology, 18(3):125–138, 2020.   
[42] Elliot J Lefkowitz, Donald M Dempsey, Robert Curtis Hendrickson, Richard J Orton, Stuart G Siddell, and Donald B Smith. Virus taxonomy: the database of the International Committee on Taxonomy of Viruses (ICTV). Nucleic Acids Research, 46(D1):D708–D717, 2018.   
[43] Daniel N Baker and Ben Langmead. Dashing: fast and accurate genomic distances with HyperLogLog. Genome Biology, 20(1):1–12, 2019.   
[44] Forest Rohwer and Rob Edwards. The Phage Proteomic Tree: a genome-based taxonomy for phage. Journal of Bacteriology, 184(16):4529–4535, 2002.   
[45] David M Kristensen, Alison S Waller, Takuji Yamada, Peer Bork, Arcady R Mushegian, and Eugene V Koonin. Orthologous gene clusters and taxon signature genes for viruses of prokaryotes. Journal of Bacteriology, 195(5):941–950, 2013.   
[46] Pakorn Aiewsakun and Peter Simmonds. The genomic underpinnings of eukaryotic virus taxonomy: creating a sequence-based framework for family-level virus classification. Microbiome, 6(1):1–24, 2018.   
[47] Soo Jen Low, Mária Džunková, Pierre-Alain Chaumeil, Donovan H Parks, and Philip Hugenholtz. Evaluation of a concatenated protein phylogeny for classification of tailed double-stranded DNA viruses belonging to the order Caudovirales. Nature Microbiology, 4(8):1306–1315, 2019.   
[48] Cynthia Maria Chibani, Anton Farr, Sandra Klama, Sascha Dietrich, and Heiko Liesegang. Classifying the unclassified: a phage classification method. Viruses, 11(2):195, 2019.   
[49] Cynthia Maria Chibani, Florentin Meinecke, Anton Farr, Sascha Dietrich, and Heiko Liesegang. Classiphages 2.0: Sequence-based classification of phages using artificial neural networks. BioRxiv, page 558171, 2019.   
[50] Benjamin Bolduc, Ho Bin Jang, Guilhem Doulcier, Zhi-Qiang You, Simon Roux, and Matthew B Sullivan. vConTACT: an iVirus tool to classify double-stranded DNA viruses that infect Archaea and Bacteria. PeerJ, 5:e3243, 2017.   
[51] Ho Bin Jang, Benjamin Bolduc, Olivier Zablocki, Jens H Kuhn, Simon Roux, Evelien M Adriaenssens, J Rodney Brister, Andrew M Kropinski, Mart Krupovic, Rob Lavigne, et al. Taxonomic assignment of uncultivated prokaryotic virus genomes is enabled by gene-sharing networks. Nature Biotechnology, 37(6):632–639, 2019.   
[52] FA von Meijenfeldt, Ksenia Arkhipova, Diego D Cambuy, Felipe H Coutinho, and Bas E Dutilh. Robust taxonomic classification of uncharted microbial sequences and bins with CAT and BAT. Genome Biology, 20(1):1–14, 2019.   
[53] Milot Mirdita, Martin Steinegger, F Breitwieser, Johannes Söding, and E Levy Karin. Fast and sensitive taxonomic assignment to metagenomic contigs. Bioinformatics, 37(18):3029–3031, 2021.   
[54] Joan Carles Pons, David Paez-Espino, Gabriel Riera, Natalia Ivanova, Nikos C Kyrpides, and Mercè Llabrés. VPF-Class: taxonomic assignment and host prediction of uncultivated viruses based on viral protein families. Bioinformatics, 2021.   
[55] Jiayu Shang, Jingzhe Jiang, and Yanni Sun. Bacteriophage classification for assembled contigs using graph convolutional network. Bioinformatics, 37(Supplement_1):i25–i33, 2021.   
[56] Yosuke Nishimura, Takashi Yoshida, Megumi Kuronishi, Hideya Uehara, Hiroyuki Ogata, and Susumu Goto. Viptree: the viral proteomic tree server. Bioinformatics, 33(15):2379–2380, 2017.   
[57] Stephen F Altschul, Thomas L Madden, Alejandro A Schäffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J Lipman. Gapped BLAST and PSI-BLAST: a new generation of protein database search programs. Nucleic Acids Research, 25(17):3389–3402, 1997.   
[58] Sean R Eddy. Accelerated profile HMM searches. PLoS Computational Biology, 7(10):e1002195, 2011.   
[59] Pascal Hingamp, Nigel Grimsley, Silvia G Acinas, Camille Clerissi, Lucie Subirana, Julie Poulain, Isabel Ferrera, Hugo Sarmento, Emilie Villar, Gipsi Lima-Mendez, et al. Exploring nucleo-cytoplasmic large dna viruses in tara oceans microbial metagenomes. The ISME journal, 7(9):1678–1695, 2013.   
[60] Jie Ren, Kai Song, Chao Deng, Nathan A Ahlgren, Jed A Fuhrman, Yi Li, Xiaohui Xie, Ryan Poplin, and Fengzhu Sun. Identifying viruses from metagenomic data using deep learning. Quantitative Biology, 8(1):64–77, 2020.   
[61] Noam Auslander, Ayal B Gussow, Sean Benler, Yuri I Wolf, and Eugene V Koonin. Seeker: alignment-free identification of bacteriophage genomes by deep learning. Nucleic Acids Research, 48(21):e121–e121, 2020.   
[62] Jiayu Shang, Xubo Tang, Ruocheng Guo, and Yanni Sun. Accurate identification of bacteriophages from metagenomic data using transformer. Briefings in Bioinformatics, 23(4), jun 2022.   
[63] Doug Hyatt, Gwo-Liang Chen, Philip F LoCascio, Miriam L Land, Frank W Larimer, and Loren J Hauser. Prodigal: prokaryotic gene recognition and translation initiation site identification. BMC Bioinformatics, 11(1):1–11, 2010.   
[64] Sergey Nurk, Dmitry Meleshko, Anton Korobeynikov, and Pavel A Pevzner. metaSPAdes: a new versatile metagenomic assembler. Genome Research, 27(5):824–834, 2017.   
[65] Christiam Camacho, George Coulouris, Vahram Avagyan, Ning Ma, Jason Papadopoulos, Kevin Bealer, and Thomas L Madden. BLAST+: architecture and applications. BMC Bioinformatics, 10(1):1–9, 2009.   
[66] Samantha Petti and Sean R Eddy. Constructing benchmark test sets for biological sequence analysis using independent set algorithms. PLoS Computational Biology, 18(3):e1009492, 2022.   
[67] Adrian Fritz, Peter Hofmann, Stephan Majda, Eik Dahms, Johannes Dröge, Jessika Fiedler, Till R Lesker, Peter Belmann, Matthew Z DeMaere, Aaron E Darling, et al. CAMISIM: simulating metagenomes and microbial communities. Microbiome, 7(1):1–12, 2019.  