# From time-series transcriptomics to gene regulatory networks: a review on inference methods  

Malvina Marku a,\* and Vera Pancaldi a,b,\* aUniversité de Toulouse, Inserm, CNRS, Université Toulouse III-Paul Sabatier, Centre de Recherches en Cancérologie de Toulouse, Toulouse, France; bBarcelona Supercomputing Center, Barcelona, Spain; \*To whom correspondence should be addressed: malvina.marku@inserm.fr , vera.pancaldi@inserm.fr  

# Abstract  

Inference of gene regulatory networks has been an active area of research for around 20 years, leading to the development of sophisticated inference algorithms based on a variety of assumptions and approaches. With the always increasing demand for more accurate and powerful models, the inference problem remains of broad scientific interest. The abstract representation of biological systems through gene regulatory networks represents a powerful method to study such systems, encoding different amounts and types of information. In this review, we summarize the different types of inference algorithms specifically based on time-series transcriptomics, giving an overview of the main applications of gene regulatory networks in computational biology. This review is intended to give an updated overview of regulatory networks inference tools to biologists and researchers new to the topic and guide them in selecting the appropriate inference method that best fits their questions, aims and experimental data.  

# Introduction  

In complex system theory, a system is defined as complex if certain properties, such as nonlinearity, feedback loops, adaptation and non-trivial behavior, emerge from the collective interactions between the system components and the surrounding environment (Artime and De Domenico, 2022). As such, all biological systems, especially molecular systems, are inherently complex, and the global structure and behavior of the system cannot be straightforwardly inferred from the (local) properties of its components. For example, characterizing the connection between the genotype and phenotype, and furthermore pathology, not only requires the identification of the molecules involved in the process and their specific characteristics, but also the ways in which these molecules interact with each other, across spatio-temporal scales. To facilitate the representation and study of such complex systems, their interacting components can be represented as a network, commonly visualized as a graph of nodes (vertices)  

connected by edges (links) (Fig. 1). In a molecular network, the nodes represent molecular objects of interest (e.g. genes, mRNAs, transcription factors (TFs), whereas the edges represent the interactions between them (e.g. protein binding, gene co-expression, TF-target regulation, etc.). Therefore, depending on the types of nodes and edges considered, different molecular networks exist, like protein-protein networks (PPI) (Koh et al., 2012; Pellegrini et al., 2004; Schwikowski et al., 2000; Vazquez et al., 2003), gene regulatory networks (hereafter, GRN) (Emmert-Streib et al., 2014; Vijesh et al., 2013), signal transduction networks (Hu et al., 2021; Kolch et al., 2015), etc.. Strictly speaking, we refer to GRNs as networks of any types of regulatory interactions between regulatory and target molecular entities (miRNAs-targets, RBP-targets, kinases/ phosphatases-substrates). Here we will focus mostly on GRNs that describe interactions between TFs and their target genes.  

In studying GRNs two main approaches of extracting information exist: (i) static network analysis, and (ii) dynamical modeling, each of which offers different amounts and types of information regarding the network organization, topology and behavior (He et al., 2015; Jungck and Viswanathan, 2015). Inferring and modeling these regulatory networks, however, is a challenging reverse engineering process and requires the combination of both a thorough biological understanding of the system, and accurate and advanced computational inference methods (Angelin-Bonnet et al., 2019). Moreover, the advanced technological improvements in measuring gene expression, in cellular populations or even single cells, and the increasing interest in clinical applications of genomics, confer importance and relevance to data-driven GRN inference methods. These approaches can ultimately provide considerable insights on gene regulation mechanisms, drugs’ mode of action, pathway perturbation, etc. than the original data alone.  

![](images/7a644e32ab8b9e3a53267bc9f4ada413d3d729353f65d8de5aff1ac920758e41.jpg)  
Fig. 1: (a) A toy model of gene regulation of three genes involved in a transcriptional regulatory network, showing genes transcribed into mRNAs and translated into proteins that regulate another one of the genes. (b) Compressed representation of the interactions on the left as a GRN in which only genes are shown with their regulatory interactions.  

With the ever continuous advances in biotechnologies, inference methods have also evolved from using bulk gene expression data (Barabási et al., 2011; Marbach et al., 2012; Silverman et al., 2020; Sonawane et al., 2019) to single-cell transcriptomics (Blencowe et al., 2019; Fiers et al., 2018; Kang et al., 2021, 2021; Matsumoto et al., 2017; Nguyen et al., 2021; Raharinirina et al., 2021a). Additionally, some methods were adapted to infer the regulatory networks from time-series and/or pseudo-temporal single-cell transcriptomics, where more accurate knowledge on gene-gene interaction can be inferred (Aubin-Frankowski and Vert, 2020a; Huynh-Thu and Geurts, 2018; Matsumoto et al., 2017).  

Consequently, high performing algorithms have been developed over the years, aiming to increase their accuracy, robustness and applicability, for example incorporating additional computational analyses, such as filtering for the presence of TF motifs in promoters of their inferred targets (Aibar et al., 2017; Alvarez et al., 2016). It is important to note, however, that GRN inference is not the final aim. Our goal in performing such calculations is to produce relevant insight on the biological processes underlying them, aiming to uncover new functionally important molecular regulatory interactions or propose new drug targets, etc (Emmert-Streib et al., 2014). Additionally, network inference offers the possibility of applying dynamical models to GRNs, in order to increase the model power to understand and predict the system temporal behavior (Angelin-Bonnet et al., 2019).  

Inference of data-driven accurate and powerful GRNs remains an open and evolving computational challenge, and new inference methods are being published continuously. In this review, we give a description of GRNs and describe the algorithms behind a selection of the available methods on network inference from experimental data. While several reviews on GRN inference based on different data sources have been published (Delgado and Gómez-Vela, 2019; Huynh-Thu and Sanguinetti, 2018; Mercatelli et al., 2020), here we will focus specifically on time- and pseudo-time series transcriptomics based inference algorithms. We conclude by giving an introduction to dynamical modeling of GRNs, as a promising tool to use GRNs in the generation of new biological hypotheses.  

# 1. Gene maps: network representations of gene regulation  

Although a single definition of GRN does not exist, we define the GRNs as topological maps representing relationships and interactions between biological entities. The interactions between proteins, transcription factors (TFs) and genes can be represented as directed graphs of nodes and edges. Notably, the directionality of the edges (defining the source and the target in the interaction) is not compulsory in all biological networks, but it represents an important feature in the case of the regulatory networks, defining the direction of information flow, as it is the case in GRNs or metabolic networks. In these networks, additional information is added by indicating specific types of interactions, represented by singed edges (Fig 1, b). In the simplest case, the interactions are categorized as activations and inhibitions, represented as positive and negative edges accordingly. GRNs are composed of regulatory nodes (source/cause nodes) and regulated nodes (target/effect nodes), generally mapped as TF-target gene network, with the incoming connectivity (in-degree) estimated using gene-centered approaches, and the outgoing connectivity (out-degree) using TF-centered approaches.  

The structure of the network enables the calculation of various quantities that capture different features of the network topology that can reveal important information on the underlying biology of the system. Centrality measures, such as eigenvector centrality, Page/Chei Rank, Burt’s constraint, or alpha centrality have been shown to highlight key nodes in a network based on network structure (Ashtiani et al., 2019, Cohen and Havlin, 2010; Newman, 2010; Newman et al., 2006). Particularly of interest, in combination with mathematical methods such as Signal Flow Analysis and Feedback Vertex Set Control (Lee and Cho, 2018), the network topology can give valuable information on the determinant nodes that define the system temporal behavior (Marazzi et al., 2022).  

# 2. Inference methods  

By definition, the process of inferring the network structure of the system, based on experimental data is a reverse engineering process, usually referred to as GRN inference. In general, based on input data used, GRN inference methods can be categorized in two types: (i) steady state gene expression, and (ii) time-series gene expression inference methods. In the first category, in principle, the GRN inference is obtained by considering perturbations of the system or in different instances and estimating the gene expression after it reaches an equilibrium. In the second category, the input data consists in gene expression measured at several time points after a perturbation thus a temporal evolution of transcriptomics can be obtained. Consequently, time series inference methods can be more informative than static data in a wide range of situations to infer gene functionalities, interactions and causal relationships, as well as to establish potential clinical implications of gene expression dynamics determined by these relationships (Bar-Joseph et al., 2012). Both methods, however, have various limitations mostly stemming from technical issues that arise from the experimental protocols: sampling time points, cost, cell synchronization, sparsity of gene expression data, etc. Therefore, several computational methods that combine both steady-state and time-series approaches have been developed, usually based on advanced machine learning algorithms. Additionally, new technology providing expression at the single-cell level have led to the development of inference methods that are specifically adapted to single-cell transcriptomics (Chen and Mar, 2018).  

Inferring the functional relationships between genes requires the estimation of gene functional dependencies, which can be broadly achieved by two different reverse engineering approaches for GRN inference:  

1. Model-free methods: In this approach, gene dependencies are inferred from using several statistical and machine learning methods, such as mutual information (Faith et al., 2007; Margolin et al., 2006; Meyer et al., 2007), random forest (Huynh-Thu et al., 2010; Kimura et al., 2020; Park et al., 2018; Petralia et al., 2015), deconvolution (Chen et al., 2014; Feizi et al., 2013), or epigenetic (Sonawane et al., 2021).  

2. Model-based methods: In this approach, a quantitative dynamical model (for example, ODEs (Aalto et al., 2020; Huynh-Thu and Geurts, 2018; Iglesias-Martinez et al., 2016), regression methods (Michailidis and d’Alché-Buc, 2013), or bayesian reasoning (Young et al., 2014)) is defined to model the dynamical properties of the system, while the regulatory network is inferred from optimizing the model parameters based on the time-series data. In this way model-based GRN methods highlight some dynamical features of the system, increasing the model interpretability, especially from a biological point of view.  

Depending on the scope and biological question for which the GRN is required, model-free or model-based inference methods can be applied; however, it is important to note that model-based methods are considerably more computationally intensive and might be limited by the underlying non-realistic linear models of gene expression dynamics. On the other hand, the Dialogue on Reverse Engineering Assessment and Methods (DREAM) project, an initiative to benchmark multiple inference methods, indicated that no single method can perform the best in every possible dataset and across different settings. Instead, a high confidence consensus network inferred from different methods is the most accurate and also provides an estimation of dataset and method robustness and performance - a process that has been usually referred as Wisdom of Crowds (Marbach et al., 2012). For this reason, instead of using a single inference method, other researchers have been developing computational tools combining several GRN inference methods (Manica et al., 2021), providing a ranking of the methods according to their performance.  

The increasing demand for improving the inference accuracy in real datasets has warranted the adaptation of some steady-state inference methods to consider time-series data. This led to the development of new promising inference methods, whose applications can highlight novel results in various molecular systems. In the next sections we will elaborate on this category of inference tools specifically and give an overview of different algorithms and their performance.  

# 2.1. Inference of GRNs from time-series transcriptomics  

Inference of gene networks from time-series data provides a more complete picture of the system than steady state data, as the biological system is intrinsically complex but biological function relies on coordinated sets of genes whose expression evolves over time. In general, inference methods based on time-series transcriptomics use a dataset containing a list of genes with their expression measured at several time points. Let’s define the dataset as $D _ { _ { T S } } .$ a matrix with dimensions $N \times T P$ , where $N$ is the number of genes and $T P$ is the number of time points at which their expression is measured:  

$$
D _ { _ { T S } } = \{ X ( t _ { _ 1 } ) , X ( t _ { _ 2 } ) , . . . , X ( t _ { p } ) \}
$$  

where $X ( t _ { p } ) , p = 1 , 2 , . . . , T P$ is a vector of $N$ genes with their expression at time $t _ { _ p }$ . The main goal of inference methods is to assign a weight $w _ { j , i } \ge 0 , i , j = 1 , 2 , . . . , N$ to any putative interaction between gene $i$ (target) and $j$ (source), representing a regulatory interaction in the biological system. To this purpose, different inference methods use various regression tools to model the expression of a gene as a function of its regulators. It is important to note that, usually, the time-series inference methods combine the time-series with steady-state datasets, in order to increase the accuracy and predictive power of the inferred network. Independent of the method chosen, the goal is to reconstruct the GRN that would produce the observed profile of expression across time, in the form of a directed graph, in which each edge is associated with its characteristic weight (Fig. 2).  

![](images/8fadc6929eaf6c8e37f88bd8b3a8421408471c4a18ff5913317b2feef6abf54e.jpg)  
Fig. 2. From time-series transcriptomics to GRN inference and model validation. The process usually starts with processing the time-series datasets and identifying the time-points relevant to describe the biological process (1). As a second step, data formatting, normalization and gene filtering and/or binarization might be necessary (2), depending on the inference method to be used (3). The inferred network consists of a weighted directed GRN (4). Model validation (5) is then necessary for quantifying the method performance in inferring the GRN when compared to gold-standard datasets, evidence from literature or prior biological knowledge. The inferred network can further be used for the application of dynamical models in cellular or multilevel modeling (6).  

# 2.2. Inference algorithms  

Based on the inference model they use, inference methods can be grouped into 7 categories, namely (i) mutual information (MI), (ii) dynamical Bayesian, (iii) Granger causality, (iv) Boolean, (v) ordinary differential equation (ODE), (vi) graphical Gaussian, and (vii) regression. It is important to note, however, that many methods apply a combination of different models and novel approaches. In the following sections, we will introduce the basic concept of each inference algorithm, while some of the available tools for using each of the methods are summarized in Table 1.  

Mutual information based models: In these methods, the putative interaction between two genes is inferred by estimating the mutual information (MI) between the expression $X _ { _ i }$ of gene $i$ at time point $t _ { _ p }$ and the expressions $X _ { _ j }$ of gene $j$ at the 𝑚 previous time points:  

$$
\begin{array} { r } { I _ { _ m } ( g _ { _ i } , g _ { _ j } ) = \displaystyle \sum _ { p = m + 1 } ^ { T P } P ( X _ { i } ( t _ { p } ) , X _ { _ j } ( t _ { p } - m ) ) l o g \frac { P ( X _ { i } ( t _ { p } ) , X _ { _ j } ( t _ { p } - m ) ) } { P ( X _ { i } ( t _ { p } ) ) P ( X _ { _ j } ( t _ { p } - m ) ) } } \end{array}
$$  

where $P ( X _ { i } ( t _ { p } ) , X _ { j } ( t _ { p } - m ) )$ is the joint probability of observing gene $i$ at time point $t _ { _ p }$ with expression $X _ { _ { i } } ( t _ { p } )$ and the expression $X _ { _ { j } } ( t _ { _ { p } } - m )$ of gene $j$ at time point $t _ { p } - m$ , whereas $P ( X _ { i } ( t _ { p } ) )$ and $P ( X _ { j } ( t _ { p } - m ) )$ are the marginal probabilities of observing gene $i$ at time point $t _ { _ { p } }$ ) with expression $X _ { _ { i } } ( t _ { p } )$ and the expression $X _ { j } ( t _ { p } - m )$ of gene $j$ at time point $t _ { p } - m$ independently. Estimating $I _ { { \bf \Pi } _ { m } } ( { \boldsymbol { g } } _ { i } , { \boldsymbol { g } } _ { j } )$ for each pair of genes and each time point, an interaction $g _ { _ j }  \ g _ { _ i }$ is being considered if $I _ { { \bf \Pi } _ { m } } ( { \boldsymbol { g } } _ { i } , { \boldsymbol { g } } _ { j } )$ exceeds a defined threshold. Among the methods using MI algorithm for GRN inference we highlight TimeDelay-ARACNE (Zoppoli et al., 2010), MRNET (Liu et al., 2017, Meyer et al., 2007) and CLR (Faith et al., 2007). One of the main advantages of these methods lies in their simplicity and speed of computation. On the other hand, their main limitation is that they do not give information on the nature of interactions (activation or inhibition) for the inferred interactions.  

Dynamical Bayesian network models: Dynamical Bayesian network model (DBN) GRN inference methods from time-series datasets (Friedman and Koller, 2003; Murphy and Mian, 1999) come as an extension of simple Bayesian network models (BN) applied to steady-state datasets, in order to overcome the limitation of prohibiting the existence of feedback loops in the network. Given that a GRN is defined as a set of vertices and edges $G = \left( V , E \right)$ , these methods search for any relationship between the expression of a target gene (defined as a random variable) and expression of its regulators (defined as parent genes) by calculating a joint probability distribution. In a time-series experiment, consisting of $N$ genes and $T P$ timepoints, the joint probability distribution of each gene is defined as:  

$$
) , . . . , X _ { N } ( t _ { 1 } ) , X _ { N } ( t _ { 2 } ) , . . . , X _ { N } ( t _ { T P } ) ) = P ( X ( t _ { 1 } ) ) \times P ( X ( t _ { 2 } ) | X ( t _ { 1 } ) ) \times . . . \times P ( X ( t _ { T P } ) | P ( X ( t _ { T P } ) ) ) .
$$  

where $X ( t _ { p } ) , p = 1 , 2 , . . . , T P$ is a vector representing the gene expression values at time point $t _ { _ p }$ . The network $G$ is then inferred by identifying the network structure with the highest posterior probability of each edge $E$ from the data, assuming a linear dependency between previous and current expression of genes. The network $G$ is Markovian, meaning that each gene is dependent only on the regulation by its regulators immediately upstream (parents). An edge is then included in the network if the marginal posterior probability of an observation (left hand side of Eq. 3 when written for each gene) exceeds a given threshold.  

Despite having a generally good performance when benchmarked against other time-series GRN inference methods (DREAM5 challenge (Marbach et al., 2012)), DBNs are generally computationally expensive and usually limited to small networks. To overcome this limit, several methods like CAS (Xing et al., 2017) or scanBMA (Young et al., 2014, implemented as a function in networkBMA R package (Fraley et al., 2014)), were developed, by combining DBN with machine learning methods or prior knowledge, to guide the search for each gene’s regulators.  

Granger causality: in these methods, two assumptions in defining the expression $X _ { _ i }$ of a gene $i$ are made: (1) the expression $X _ { _ i }$ of gene $i$ at time point $t _ { _ p }$ is a function of its expression at the $m$ previous time points only, (2) the expression $X _ { _ i }$ of gene $i$ at time point $t _ { _ { p } }$ is a function of both its expression at the $m$ previous time points and the expression $X _ { _ j }$ of genes $j$ at the $m$ previous time points, thus indicating the effect of gene $j$ on gene $i$ . If the second assumption is significantly more successful than the first assumption, it is said that gene $j$ Granger causes gene $i$ .  

$$
\begin{array} { r l } & { X _ { i } ( t _ { p } ) = \alpha _ { 0 } + \alpha _ { 1 } X _ { i } ( t _ { p } - 1 ) + \ldots + \alpha _ { m } X _ { i } ( t _ { p } - m ) + \eta ( t _ { p } ) ( 4 \mathsf { a } ) } \\ & { = \alpha _ { 0 } + \alpha _ { 1 } X _ { i } ( t _ { p } - 1 ) + \ldots + \alpha _ { m } X _ { i } ( t _ { p } - m ) + \beta _ { 0 } + \beta _ { 1 } X _ { j } ( t _ { p } - 1 ) + \ldots + \beta _ { m } X _ { j } ( t _ { p } - m ) } \\ & { \qquad i , j = 1 , 2 , . . . , N \qquad p = 1 , 2 , . . . , T P } \end{array}
$$  

where $\alpha , \beta$ are coefficients and $\boldsymbol \eta ( t _ { p } )$ is the residual noise at time $t _ { _ p }$ . Generalizing for $N$ genes, gene $j \left( j = 1 , 2 , . . . , N , j \neq i \right)$ is said to be casual for gene $i$ if considering the expression of gene $j$ in the previous time points significantly improves the prediction of the expression of gene $i$ at the current time point. Notably, when referring to the ‘previous time points’, several previous time points $( t _ { p } - 1 , t _ { p } - 2$ , etc) can be considered. Considering $N$ genes in the dataset, the vector autoregressive $( V O A )$ model is used to estimate the Granger causality over the entire list of genes, where a linear dependency between genes is assumed (Tam et al., 2013).  

Several methods using Granger causality for GRN network inference have been developed, such as BETS (Lu et al., 2021), SWING (Finkle et al., 2018), CGC-2SPR (Yao et al., 2015), among others. One of the notable advantages of this approach is the computational efficiency and speed, compared to other inference methods, with a similar performance. However, considering their definitions and associated algorithms, Granger causality inference methods can be used only on time-series datasets, which are often limited by the sparsity and non-uniformity of time-point spacing. For this reason, the method was extended to consider a flexible time lag between consecutive time-points (Finkle et al., 2018).  

Boolean models: in these methods, a GRN is represented as a graph $G = \left( V , E \right)$ and $< i , j , s >$ indices, in which $i$ represents the target and $j$ represents the source gene. Each edge is characterized by a sign $s \in \{ + , - \}$ indicating the type of interaction between the regulator and the source node. Generally, a positive sign indicates a positive (activating, promoting) relationship between a regulator and its target gene (i.e the regulator gene $j$ contributes in increasing the expression of target gene $i$ through specific biological processes), whereas a negative sign indicates a negative (inhibiting, repressing) relationship (i.e the more expressed the regulator gene $j$ is, the less expressed the target gene $i$ will be). Notably, the expression of each gene is given in Boolean (binary) values {0, 1}, thus providing only a qualitative description of gene regulation and expression. Given a time-series dataset, the expression $X _ { _ i }$ of gene $i$ at time $t _ { _ p }$ is given by a Boolean function of its regulators:  

$$
X _ { i } ( t _ { p } ) = F _ { _ { B _ { i } } } ( X _ { i , 1 } ( t _ { p } - 1 ) , X _ { i , 2 } ( t _ { p } - 1 ) , . . . , X _ { _ { i , k } } ( t _ { p } - 1 ) )
$$  

where $X _ { i , k } , \ k = 1 , 2 , . . . , N - 1 , k \neq i$ are the regulators of gene $i$ , and $F _ { _ B }$ is a Boolean function describing the regulation of gene $i$ using the Boolean operators AND, OR and NOT. In this way, the inference of any causal relationship is done by a discrete dynamical model of gene expression (Kauffman and Kauffman, 1993; Shmulevich et al., 2002a).  

Compared to the other inference methods, BN inference methods offer the advantage of being parameter free in the modeling of gene regulation and relatively easy to apply. On the other hand, they pose several difficulties and limitations arising from the discretization/binarization process, the finite space explored for inferring the regulatory processes (limited number of possible BNs), and the qualitative description of expression (Pušnik et al., 2022). Nevertheless, several inference methods based on BN have been developed, such as REVEAL (Liang et al., 1998) and Best-Fit (Lähdesmäki et al., 2003; Shmulevich et al., 2002b), both available as extensions in the BoolNet R package (Müssel et al., 2010), and ATEN (Shi et al., 2019).  

Ordinary differential equation (ODE) models: in this formalism, the variation of expression $X _ { _ i }$ of gene $i$ at time $t _ { _ p }$ is given by a nonlinear function of its regulators (including self regulation) at the same time steps:  

$$
\begin{array} { r } { \frac { d X _ { i } } { d t } = \displaystyle \sum _ { k = 1 } ^ { N - 1 } f ( X _ { i , k } ) + \boldsymbol { \eta } _ { i } } \end{array}
$$  

where the functions $f ( X _ { i , k } )$ are usually assumed to be polynomial functions (Kim et al., 2007) and $\boldsymbol { \mathfrak { n } } _ { i }$ represents the non-deterministic term (noise) in the regulatory function of $X _ { _ i }$ . The regulatory network is inferred solving a system of ODEs (one equation for each gene). From its definition, solving the ODE system (Eq. 6) is computationally and conceptually challenging, even in a linear space, as these systems are defined by a large number of interacting coefficients, for which the prior information is limited or missing. Consequently, ODE methods are limited to inference of small regulatory networks and are usually combined with other inference methods in order to reduce the computational complexity (Margolin et al., 2006; Meyer et al., 2007). In principle, a causal relationship between two genes is considered if the interaction is described by a relatively high interacting coefficient. Despite the difficulty of applying them to real datasets, several methods using ODE models have been developed, including Inferelator (Bonneau et al., 2006) and TSNI (Bansal et al., 2006), but they often display a relatively lower performance than alternative approaches, as reported in (Lu et al., 2021).  

Regression models: generally, in these methods a non-linear regression or ODE formulation is used to model the expression of a gene at a certain time point/condition as a nonparametric function of the expression of other genes at the same time point/condition:  

$$
X _ { _ i } = f _ { _ i } ( X _ { _ j } ( t _ { p } ) ) \ + \ \mathfrak { \eta } ( t _ { p } ) ,
$$  

where $\boldsymbol { \mathfrak { \boldsymbol { \eta } } } ( t _ { p } )$ is a random noise term for the ${ t _ { p } ^ { \ t h } }$ time point. The inference of $f _ { _ i }$ functions is then done by implementing a feature selection algorithm, like LASSO, least angle regression (LARS), or random forest algorithm to “learn” these functions from an ensemble of regression trees (Breiman, 2001). To identify the candidate genes for the regulation of $N$ genes, this approach starts by splitting the problem into $N$ distinct sub-problems, thus assuming that each gene can be a target gene and a potential regulator. Causal interactions are quantified by estimating the confidence levels of each $j \to i$ interaction, which is represented by the weights $w _ { j , i }$ . Importantly, in order to increase the accuracy of the inferred network, some regression methods have been adapted to use both time-series and steady state datasets. Some inference methods based on decision-tree algorithms can be highlighted to have particularly good performance: dynGENIE3 (Huynh-Thu and Geurts, 2018) and Jump3 (Huynh-Thu and Sanguinetti, 2015), both being extensions of the steady-state method GENIE3 (Huynh-Thu et al., 2010) for time-series datasets.  

Gaussian process (GP) models: in these methods, a GP regression is used to model the relationship between the current expression of the genes and their previous expression levels, modeling the functional relationships between targets and their regulators as Gaussian processes:  

$$
X _ { i } ( t _ { p } ) = f ( X _ { j } ( t _ { p } ) ) + \eta ( t _ { p } )
$$  

A causal relationship (edge) between two genes is considered based on the sum of its posterior probabilities of the existence of this relationship (edge) over the time points. In this way, Gaussian process models are generally considered as nonlinear DBN. Consequently, the derived inference methods, such as the algorithm presented by Aijo and Lähdesmäki (Aijö and Lähdesmäki, 2009) and BINGO (Aalto et al., 2020) combine both DBN and GP.  

# 2.3. Inference from time-series of single-cell transcriptomics  

For some biological processes, like cell differentiation, phenotypic reprogramming, etc, a higher resolution of the temporal dynamics of gene expression is necessary to identify major phenotypic transitions in complex tissues while characterizing the phenotypic spectrum of individual cells. In this regard, single-cell RNA-sequencing technology enables deeper investigation of the molecular interactions and identification of novel molecular mechanisms that orchestrate biological processes at the single cell level. Computationally, this technological revolution has led to the development of several algorithms to analyze single-cell RNA-seq, and - as a part of it - inference of GRNs (Fig. 3). Intuitively, a single-cell resolution of genes’ dynamics would lead to an increased accuracy in inferring the functional interactions between genes that define the biological process. However, dealing with limitations in some of the most widely available single-cell technologies, the heterogeneity and sparsity of single-cell data lead to limitations and challenges for GRN inference methods and put reliability in question.  

An important point in using single-cell RNA-seq data for GRN inference is - usually - the lack of time-resolved expression measurements. Instead, many inference methods exploit the multiplicity of RNAseq profiles at one single time point across cells as a proxy for temporal evolution of the phenotype, as is the case for trajectory inference based on pseudo-time ordering of the cells (Aibar et al., 2017), assuming ergodicity of the phenotypes.  

Table 1: Tools for GRN inference from time-series transcriptomics categorized by their inferring algorithm. The characteristics of the inferred network are indicated as follows: $0$ undirected, $D$ directed and unsigned, ▶ directed and signed.   


<html><body><table><tr><td>Tool</td><td></td><td>Method</td><td>Language</td><td>Reference</td></tr><tr><td>Time-delayed ARACNE</td><td></td><td>Mutual Information</td><td>R</td><td>(Zoppoli et al., 2010)</td></tr><tr><td>MRNET</td><td></td><td>Mutual Information</td><td>minet R package</td><td>(Liu et al., 2017)</td></tr><tr><td>CLR</td><td>□</td><td>Mutual Information</td><td>minet R package</td><td>(Faith et al., 2007)</td></tr><tr><td>Jump3</td><td>□</td><td>Regression</td><td>Matlab/ R</td><td>(Huynh-Thu and Sanguinetti, 2015)</td></tr><tr><td>dynGENIE3</td><td></td><td>Regression (+ ODE)</td><td>R</td><td>(Huynh-Thu and Geurts, 2018)</td></tr><tr><td>SWING-RF</td><td>□</td><td>Granger causality + Regression</td><td>Python</td><td>(Finkle et al., 2018)</td></tr><tr><td>BETS</td><td></td><td>Granger causality</td><td>Python</td><td>(Lu et al., 2021)</td></tr><tr><td>CGC-2SPR</td><td>□</td><td>Granger causality</td><td>R/Matlab</td><td>(Yao et al., 2015)</td></tr><tr><td>CAS</td><td></td><td>Dynamical Bayesian Model</td><td>No info</td><td>(Xing et al., 2017)</td></tr><tr><td>scanBMA</td><td>□</td><td>Dynamical Bayesian Model</td><td>R</td><td>(Young et al., 2014)</td></tr><tr><td>Package GeneNet</td><td>□</td><td>Dynamical Bayesian Model</td><td>R</td><td>(Kolpakov et al., 1998)</td></tr><tr><td>REVEAL</td><td></td><td>Boolean Network + Mutual Information</td><td>C</td><td>(Liang et al., 1998)</td></tr><tr><td>BoolNet</td><td>□</td><td>Boolean Network</td><td>R</td><td>(Mussel et al., 2010)</td></tr><tr><td>GABNI</td><td></td><td>Boolean Network + Ml + Genetic Algorithm</td><td></td><td>(Barman and Kwon, 2018)</td></tr><tr><td>ATEN</td><td></td><td>Boolean Network + Regression</td><td>R</td><td>(Shi et al., 2019)</td></tr><tr><td>TSNI</td><td></td><td>ODE</td><td>Matlab</td><td>(Bansal et al., 2006)</td></tr><tr><td>Inferelator</td><td></td><td>ODE</td><td>R</td><td>(Bonneau et al., 2006)</td></tr></table></body></html>  

<html><body><table><tr><td>BINGO</td><td>V</td><td>Gaussian process</td><td>Matlab</td><td>(Aalto et al., 2020)</td></tr></table></body></html>  

In this process, the pseudo-temporal trajectory is generated by linearly ordering the single-cell profiles from a specific time point based on their transcriptional similarity, thus enabling the identification of gene patterns along the developmental trajectory of continuously ordered cells (Zeng et al., 2017). Accordingly, a subset of the inference methods require specific information about the pseudotemporal ordering of the cells (SCODE (Matsumoto et al., 2017), SINCERITIES (Papili Gao et al., 2018), SINGE (Deshpande et al., 2022), LEAP (Specht and Li, 2016), SCRIBE (Qiu et al., 2020), etc.), having a significant difference in performance when such information is not available. Other methods, like GENIE3 (Huynh-Thu et al., 2010), GRNBoost2 (Moerman et al., 2019), or PPCOR (Kim, 2015) do not require a temporal ordering of the cells as input and have relatively good performance when tested on some published curated models (Pratapa et al., 2020). However, the incomplete equivalence between bulk gene expression time courses and pseudotime time-series implies that these two types of inference cannot be always performed by the same tools.  

Several benchmarking papers on the performance of single-cell RNA-seq inference methods have been published, facilitating the benchmarking of different inference methods. We refer the reader to (Blencowe et al., 2019; Fiers et al., 2018; Nguyen et al., 2021) for an extensive review and comparison of single-cell RNA-seq inference methods, (Bellot et al., 2015; Kang et al., 2021; Pratapa et al., 2020) for some benchmarking libraries and to (Huynh-Thu and Sanguinetti, 2015) for an algorithmic review. In principle, these methods follow a similar classification as in bulk transcriptomics, described in Section 2.2, as many of them have been adapted for usage in single-cell from previous existing methods for bulk data. For this reason, here we simply give a list of some of these methods (Table 2), noting that there are many more and the list is increasing rapidly.  

![](images/2dcb4708d4fed08f23a636659f7c04265dc579327dbcd78a3af128f313abd57f.jpg)  
Fig. 3: GRN inference from single-cell (pseudo)time- series. The inference workflow follows a similar path as in bulk time-series transcriptomics, except from additional steps of dimensionality reduction and trajectory inference (2), and pseudotime ordering of the cells (3) when time-resolved experimental measurements are not available. Steps (4) - (6) follow the same logic as in Fig. 2.  

For a user, the choice between all the different inference methods will depend on both their overall performance, the type and amount of information they require, and the type of reconstructed network they provide. For example, SCODE, PPCOR and SINCERITIES infer a directed and signed GRN, which can further be used easily to make dynamical models, if necessary.  

Table 2: GRN inference tools from single-cell transcriptomics categorized by their inferring algorithm. The characteristics of the inferred network are indicated as follows: $0$ undirected, $D$ directed, $\blacktriangleright$ directed and signed.   


<html><body><table><tr><td>Tool</td><td></td><td> Method</td><td>Language</td><td>Reference</td></tr><tr><td>SINCERITIES</td><td>□</td><td>Correlation ensemble</td><td>R/Matlab</td><td>(Papili Gao et al., 2018)</td></tr><tr><td>GENIE3</td><td>□</td><td>ODE + Regression</td><td>R</td><td>(Huynh-Thu et al., 2010)</td></tr><tr><td>GRNBoost2</td><td>□</td><td>Regression</td><td>Python</td><td>(Moerman et al., 2019)</td></tr><tr><td>PPCOR</td><td>□</td><td>Semi-partial correlation</td><td>R</td><td>(Kim, 2015)</td></tr><tr><td>LEAP</td><td></td><td>Correlation</td><td>R</td><td>(Specht and Li, 2016)</td></tr><tr><td>CARDAMOM* WASABI HARISSA</td><td></td><td>Regression + wave propagation</td><td>Python</td><td>(Ventre et al., 2022) (Bonnaffoux et al., 2019) (Herbach et al., 2017)</td></tr><tr><td>SINGE</td><td>□</td><td>Granger causality</td><td>Matlab</td><td>(Deshpande et al., 2022)</td></tr><tr><td>AR1MA1 - VBEM</td><td>?</td><td>Bayesian Dynamics</td><td>Matlab</td><td>(Sanchez-Castillo et al., 2018)</td></tr><tr><td>GRISLI</td><td>□</td><td>ODE</td><td>Matlab</td><td>(Aubin-Frankowski and Vert, 2020b)</td></tr><tr><td>SCODE</td><td>□</td><td>ODE</td><td>R/Julia/Ruby</td><td>(Matsumoto et al., 2017)</td></tr><tr><td>SCRIBE</td><td></td><td>Mutual Information</td><td>C++/R</td><td>(Qiu et al., 2020)</td></tr><tr><td>PIDC</td><td>①</td><td>Mutual Information</td><td>Julia</td><td>(Chan et al., 2017)</td></tr><tr><td>Boolean Pseudotime</td><td></td><td>Boolean Model</td><td>Python</td><td>(Hamey et al., 2017)</td></tr><tr><td>InferenceSnaps hot</td><td></td><td>Boolean Model</td><td>C++/Matlab</td><td>(Ocone et al., 2015)</td></tr></table></body></html>

\* Combined methods  

# 3. GRN validation  

A crucial and essential step in performing GRN inference is validation of the inferred network. Having a validation protocol that allows evaluation of the score of each proposed model is very important in order to choose the optimal inference procedure among the variety of existing algorithms. Despite advances in this direction, evaluating the effectiveness of the inference method remains an open challenge, mostly due to limitations in ground truth/gold standard datasets. In some cases, available networks from different public databases, such as RegulonDB (Salgado et al., 2013), KEGG (Kanehisa et al., 2008), ESCAPE (Xu et al., 2013) etc, are used as ground-truth (or reference) networks, and a comparison between the inferred and the reference gold-standard networks is performed.  

Particularly, the ESCAPE database (Xu et al., 2013) represents one of the most comprehensive repositories for ChIP-seq - perturbation data. However, the repository contains reference data for a limited number of interactions, limiting the validation to a subset of the inferred network. This implies that the interactions for which the reference data is absent are considered as not existing, raising important questions on the implication of previous biases or the potential to infer novel interactions and regulators. Generally, GRN inference methods aim to infer novel interactions and regulators that might lead to identifying new regulatory pathways involved in the system under consideration, for which gold standard references with high scores are absent or sparse. Another possibility for network validation comes from using simulated data, which can be engineered to include several conditions and measurements (Marbach et al., 2012) - yet the extent of  coverage of the inferred interactions remains limited for small network size.  

Other than validating the inferred network with a gold-standard one, another important point in GNR inference is network comparison between different inference methods, on the same dataset, thus providing an estimate of the “robustness” of the inferred network. Additionally, one may perform this benchmark analysis in order to choose the algorithm/method that best suits a given dataset.  

Quantitatively, the algorithm’s performance can be evaluated as for any prediction by two metrics: (1) area under the precision-recall curve, estimating the performance of a certain algorithm, and (2) area under receiver operating characteristic curve, comparing the GRN inferred by the algorithm against a gold standard network.Considering every inferred interaction in the GRN as a positive (true - TP, or false - FP) and any missing one as negative (true - TN, or false - FN), the two measures are defined as:  

1. Area under precision-recall (AUPR): starting from the inferred weighted GRN, one might start by filtering by the highest values of the weights $( w _ { i j } \in [ 0 , 1 ] )$ , for which the precision (prediction) will be high and recall (retrieved) will be low. Adding interactions with lower weights will lead to a decrease of precision, until at $\mathsf { r e c a l l } = 1$ (no filter, fully connected network) the precision will represent the fraction of true positives over the total number of edges in the fully connected network. Graphically, this procedure is represented by a precision-recall curve: a good performing algorithm will have a high precision even with increasing the recall, and an area under precision-recall close to 1 (Fig. 4 (a)). Generally, this measure is used as a global performance estimator for most of the inference algorithms.  

2. Area under the receiver operating characteristic (AUROC) curve score, defined as the ratio between the true positives and false positives, graphically represented with the TP on the 𝑦 axis and the FP on the $x$ axis (Fig. 4 (b)).  

In addition, the structural features of inferred networks, such as PageRank, heat diffusion, the shortest path, etc., can be used to compare GRNs inferred from different methods (Arici and Tuncbag, 2021), whereas other metrics can be used to define the method performance, such as stability (across simulations, artificially removing measurements), identification of network motifs, and computational time and memory usage (Pratapa et al., 2020).  

![](images/9b14bac159f38810f73c63a6a86bde1d58acbdda669a6bd7aea02408d004a168.jpg)  
Fig. 4: Example of AUPR (a) and AUROC (b) curves. The graphs are obtained using PRROC R-package (Grau et al., 2015).  

An important question arises when comparing the structure of the same cellular network of two different cellular states, e.g different phenotypes of a certain cell or healthy vs disease conditions. This comparative analysis, referred to as differential network analysis (DiNA), remains a challenge, in parallel with the development of GRN inference methods, especially since different data types other than RNAseq can be included in GRN inference in order to increase the accuracy of the inferred network. For example, using single-cell ATAC-seq, patterns of gene expression can be combined with chromatin accessibility profiles, thus identifying cell subpopulations and groups of cells at different developmental stages (Duren et al., 2021). Consequently, several differential network analysis algorithms have been developed, such as DiffRank (Odibat and Reddy, 2011), dcanr R package (Bhuva et al., 2019), DiffK (Fuller et al., 2007), DINA (Gambardella et al., 2013), to name a few. We refer the reader to (Shojaie, 2021) for a statistical perspective of DiNA and (Lichtblau et al., 2017) for a DiNA algorithm comparison. In principle, these algorithms consist of combining the information from two main computational tasks: (i) differential expression analysis, which estimates the differential gene expression abundance for each gene in the GRN between the two conditions, and (ii) network expression analysis, which estimates the importance of each gene in the GRN, based on the topological properties of the network. In this way, DiNA algorithms aim at identifying genes or subnetworks of genes whose expression changes the most across conditions and they are especially suitable in cases when the changes in the network structure lead to phenotypic changes in the system. Therefore, numerous publications have demonstrated the power of this analysis, such as for identifying key TFs involved in cancer, when compared to healthy controls (Duren et al., 2021).  

# 4. Applications  

Although inferring regulatory networks is a trending topic and despite the multitude of different algorithms available, GRN inference methods struggle to reach a high performance in real-world studies, on both bulk and single-cell RNA-seq data, as reported in (Chen and Mar, 2018). Therefore their application to biologically relevant datasets remains limited. Nevertheless, some applications of inferring GRNs for novel discoveries in biology have been presented in most of the cited works on GRN inference methods. Other applications include building molecular disease maps (Janssens et al., 2022; Van Hove et al., 2019), phenotypic characterization of a cell in a given microenvironment (Lambrechts et al., 2018; Patsalos et al., 2021), identifying predictive or prognostic biomarkers (Lu et al., 2021), discovering new therapeutic targets/regulators (Hossain et al., 2021), performing extensive studies on performance of the available methods on different datasets/conditions (Duren et al., 2021; Jackson et al., 2020; Raharinirina et al., 2021b), and many more.  

# 5. From static to dynamic networks  

Interestingly, inferred GRNs can serve as a backbone for constructing dynamical models: one can start from a temporal analysis of the system behavior under multiple conditions, describing how the abundances of the genes in the network change due to their regulatory interactions. Dynamical models of GRNs cover a spectrum as wide as the GRN inference methods do. Many dynamical models exist, ranging from continuous quantitative ODE models to discrete logic quantitative models (Fig. 5, (a)). In continuous models, the temporal evolution of the state $X _ { _ i }$ of gene $i ( i = 1 , 2 , . . . , N )$ in the network is given by a continuous ODE function of its regulators. In this way, the regulatory interactions between genes are modeled as chemical reaction models, or species interactions in ecological models (Polynikis et al., 2009; Takeuchi, 1996). Beneficially, these models can capture the temporal evolution of the system at the level of individual reactions. However, their usage in dynamical modeling of GRN remains limited because of the detailed and complete mathematical and parametric description they require. On the other side of the spectrum, in discrete models, such as multivariate logic models (Aldridge et al., 2009), Petri nets (Murata, 1989), or Boolean models (Glass and Kauffman, 1973; Kauffman and Kauffman, 1993), a discrete logic of interactions is applied and the temporal evolution of expression of each gene is given by a discrete (and often logic) function of their regulators. Contrary to continuous models, discrete models can be applied with no or considerably fewer parameters and fragmented mechanistic description, making them suitable for large networks. However, the system behavior is only described (semi)quantitatively. Other methods use the network structure to infer the asymptotic behavior of the system, i.e, identifying only the steady states of the system, considering only the static GRN and a set of initial conditions, without requiring a dynamical description (Lee and Cho, 2018; Marazzi et al., 2022). For a modeler, the choice of the modeling method is going to depend on the type of questions being asked on the system, the type of available description of the system dynamics (quantitative or qualitative), the type and amount of available data, prior knowledge, etc. Whichever the dynamical model used, the main goal is to identify the phenotypic changes of a cell in response to certain extracellular environmental conditions, drugs, to cellular interactions in the microenvironment, or in experimental knockout/overexpression experiments. Graphically, these phenotypic changes can be identified as the system’s steady states or attractors (fixed points or limit cycles) (Fig. 5 (b) (Shah et al., 2018), defined as a state (a vector of expression values of each gene), which remains unvaried even in the presence of perturbations. Usually, a GRN can have multiple attractors, representing the possible phenotypes or cell states that can be reached when starting from the given initial conditions. In this case, further analysis must be performed on the attractors, to study their stability, their biological relevance, or - additionally - their categorization into known phenotypes.  

![](images/0d655d7913c9107f7c312633ae4896ffea53128042a397e8d41e07bc8fb3d74a.jpg)  
Fig. 5: (a) Dynamical models of gene regulatory networks, represented as a spectrum of models ranging from continuous to discrete. (b) Steady states of a system composed of 3 genes: (left) ODE model, as a system of 3 interacting species, (right) Boolean model, as logic-based interacting entities.  

Particularly in Boolean models, similar to using time-series of expression data for GRN inference, the temporal information in the changes of the genes’ expression is also used to infer the Boolean rules that govern these changes (Barman and Kwon, 2018; Gjerga et al., 2020; Hall and Niarakis, 2021; Henao et al., 2022; Ostrowski et al., 2016; Razzaq et al., 2018), which can be then studied using several available tools (Naldi et al., 2018). More recently, patient-specific Boolean models have been developed to design targeted therapy strategies for patients based on their omics profile (Montagud et al., 2022). Importantly, these intracellular models can be further integrated with other cell population models, like agent-based models or metabolic models, thus providing a multilevel description of the system dynamics, including mechanistic functionalities like cell motility, cytokine diffusion, tissue expansion and spatial organization, etc. (Aguilar et al., 2020; Letort et al., 2019; Stoll et al., 2022). This combined approach enables addressing more complex questions, like drug design, or therapy action from the cell to the tissue scale.  

# 6. Discussion - open challenges  

In this review we give a basic introduction to GRNs, their topological characteristics and, most importantly, describe the main GRN inference methods. Our aim is to give life scientists an overview of how to use the abstract concept of GRNs to investigate complex cellular molecular interactions more thoroughly and identify how specific interactions determine cells’ behavior and response to the environment. With the increasing abundance of transcriptomics data, data-derived GRNs have the potential to capture novel gene interactions and help to expand our knowledge of important molecular pathways. Despite the increasing interest in developing high performing GRN inference methods and this field of research having been active for more than 20 years, often the application of even the best performing methods in real-world studies raises questions about their reliability and purely data-driven GRN inference remains still an open challenge, especially in single-cell RNAseq. Recent promising research addresses the challenge of network inference in the presence of incomplete multi-omics datasets, resulting in the development of advanced computational methods (Henao et al., 2022). We speculate that perhaps the information provided from transcriptomics data, even in single-cell, is not sufficient to cover the complex cellular processes giving rise to phenotypes, and that the inherited concept that co-expression patterns might reveal putative gene interactions might not be universally applicable. For example, due to post-transcription and translation processes, a highly expressed mRNA may not lead to a functional protein, and vice versa - a synthesized protein does not necessarily possess the necessary activation state or conformation or localization to affect its downstream targets. Therefore, there is a necessity to produce multiple types of omics data and the challenge in performing GRN inference will be in how to integrate the data effectively.  

Additional limitations in the performance of GRN inference methods can come from other - often ignored - sources. For example, gathering experimental samples from various patients can hinder the variability characterizing patients of individual history, immune system, or genetics. Experimental protocols can present restrictions as well, in limited measurements of expression (transcription, degradation, sequencing capture, etc.), heterogeneity in bulk datasets or incompleteness in single-cell RNAseq transcriptomics.  

Another challenge in data-driven GRN inference is their interpretability and the difficulty in dealing with the high complexity increasing with the network size. During the cell state transitions (like differentiation or polarization) in a multicellular system, we can imagine that the interactions between TFs and their target genes can be cell-type specific, thus representing the regulatory network of a specific cell. However, all of the inference methods produce a single  

GRN thus failing to provide the dynamics of the underlying mechanisms during cell state transition during the time-series. One possibility in dealing with this challenge is to infer stage-specific GRNs, thus having a time-evolving GRN which can help understanding how the involvement and interactions of specific genes/TFs lead to cell state transition. However, studying and understanding time-evolving GRNs remains an unexplored field of high complexity. Additionally, tracking specific regulatory pathways or interactions rapidly becomes a real challenge when dealing with large networks consisting of thousands of genes. We assume that topological network analysis might help in reducing the network size by selecting the most influencing nodes (by their centrality measures), although there is still a gap in our understanding between the structural and dynamical properties of the network (Luscombe et al., 2004; Klamt et al., 2006; Barabási et al., 2011; Yang et al., 2018).  

One of the most difficult challenges in performing data-driven GRN inference is validating the results and estimating the method performance in real-case studies. Benchmarking the inference methods with simulated datasets from a prior knowledge network (PKN) and a small list of genes can be an efficient way to estimate the methods’ performance. But this is far from real-case studies, where the actual regulatory mechanisms are mostly unknown and - in some cases - experimentally unexplored. Therefore using public databases (usually built on bulk-RNAseq) for validating the inferred GRNs can be limited to small-size GRNs and narrowed around the highly studied regulatory pathways. Consequently, this leads to biased conclusions and a real difficulty in identifying novel regulatory pathways that might play an important role in the system under study.  

We believe that many of these challenges will be addressed in the new inference methods to be developed in the future. Despite the considerable improvements and the rapid growth in numbers of GRN inference methods, this remains a relatively new and highly complex field. Feeding the methods with different types of ‘omics data and prior knowledge, GRN inference can help discover unknown pathways, biological components and interactions, thus further increasing our knowledge, in a positive feedback loop.  

# References  

Aalto, A., Viitasaari, L., Ilmonen, P., Mombaerts, L., Gonçalves, J., 2020. Gene regulatory network inference from sparsely sampled noisy data. Nat. Commun. 11, 3493. https://doi.org/10.1038/s41467-020-17217-1   
Aguilar, B., Gibbs, D.L., Reiss, D.J., McConnell, M., Danziger, S.A., Dervan, A., Trotter, M., Bassett, D., Hershberg, R., Ratushny, A.V., Shmulevich, I., 2020. A generalizable data-driven multicellular model of pancreatic ductal adenocarcinoma. GigaScience 9, giaa075. https://doi.org/10.1093/gigascience/giaa075   
Aibar, S., González-Blas, C.B., Moerman, T., Huynh-Thu, V.A., Imrichova, H., Hulselmans, G., Rambow, F., Marine, J.-C., Geurts, P., Aerts, J., van den Oord, J., Atak, Z.K., Wouters, J., Aerts, S., 2017. SCENIC: Single-cell regulatory network inference and clustering. Nat. Methods 14, 1083–1086. https://doi.org/10.1038/nmeth.4463   
Aijö, T., Lähdesmäki, H., 2009. Learning gene regulatory networks from gene expression measurements using non-parametric molecular kinetics. Bioinforma. Oxf. Engl. 25, 2937–2944. https://doi.org/10.1093/bioinformatics/btp511   
Aldridge, B.B., Saez-Rodriguez, J., Muhlich, J.L., Sorger, P.K., Lauffenburger, D.A., 2009. Fuzzy Logic Analysis of Kinase Pathway Crosstalk in TNF/EGF/Insulin-Induced Signaling. PLOS Comput. Biol. 5, e1000340. https://doi.org/10.1371/journal.pcbi.1000340   
Alvarez, M.J., Shen, Y., Giorgi, F.M., Lachmann, A., Ding, B.B., Ye, B.H., Califano, A., 2016. Network-based inference of protein activity helps functionalize the genetic landscape of cancer. Nat. Genet. 48, 838–847. https://doi.org/10.1038/ng.3593   
Angelin-Bonnet, O., Biggs, P.J., Vignes, M., 2019. Gene Regulatory Networks: A Primer in Biological Processes and Statistical Modelling. Methods Mol. Biol. Clifton NJ 1883, 347–383. https://doi.org/10.1007/978-1-4939-8882-2_15   
Arici, M.K., Tuncbag, N., 2021. Performance Assessment of the Network Reconstruction Approaches on Various Interactomes. Front. Mol. Biosci. 8, 666705. https://doi.org/10.3389/fmolb.2021.666705   
Artime, O., De Domenico, M., 2022. From the origin of life to pandemics: emergent phenomena in complex systems. Philos. Trans. R. Soc. Math. Phys. Eng. Sci. 380, 20200410. https://doi.org/10.1098/rsta.2020.0410   
Ashtiani, M., Mirzaie, M., Jafari, M., 2019. CINNA: an R/CRAN package to decipher Central Informative Nodes in Network Analysis. Bioinformatics 35, 1436–1437. https://doi.org/10.1093/bioinformatics/bty819   
Aubin-Frankowski, P.-C., Vert, J.-P., 2020a. Gene regulation inference from single-cell RNA-seq data with linear differential equations and velocity inference. Bioinformatics 36, 4774–4780. https://doi.org/10.1093/bioinformatics/btaa576   
Aubin-Frankowski, P.-C., Vert, J.-P., 2020b. Gene regulation inference from single-cell RNA-seq data with linear differential equations and velocity inference. Bioinformatics 36, 4774–4780. https://doi.org/10.1093/bioinformatics/btaa576   
Bansal, M., Gatta, G.D., di Bernardo, D., 2006. Inference of gene regulatory networks and compound mode of action from time course gene expression profiles. Bioinformatics 22, 815–822. https://doi.org/10.1093/bioinformatics/btl003   
Barabási, A.-L., Gulbahce, N., Loscalzo, J., 2011. Network medicine: a network-based approach to human disease. Nat. Rev. Genet. 12, 56–68. https://doi.org/10.1038/nrg2918   
Bar-Joseph, Z., Gitter, A., Simon, I., 2012. Studying and modelling dynamic biological processes using time-series gene expression data. Nat. Rev. Genet. 13, 552–564. https://doi.org/10.1038/nrg3244   
Barman, S., Kwon, Y.-K., 2018. A Boolean network inference from time-series gene expression data using a genetic algorithm. Bioinformatics 34, i927–i933. https://doi.org/10.1093/bioinformatics/bty584   
Bellot, P., Olsen, C., Salembier, P., Oliveras-Vergés, A., Meyer, P.E., 2015. NetBenchmark: a bioconductor package for reproducible benchmarks of gene regulatory network inference. BMC Bioinformatics 16, 312. https://doi.org/10.1186/s12859-015-0728-4   
Bhuva, D.D., Cursons, J., Smyth, G.K., Davis, M.J., 2019. Differential co-expression-based detection of conditional relationships in transcriptional data: comparative analysis and application to breast cancer. Genome Biol. 20, 236. https://doi.org/10.1186/s13059-019-1851-8   
Blencowe, M., Arneson, D., Ding, J., Chen, Y.-W., Saleem, Z., Yang, X., 2019. Network modeling of single-cell omics data: challenges, opportunities, and progresses. Emerg. Top. Life Sci. 3, 379–398. https://doi.org/10.1042/ETLS20180176   
Bonnaffoux, A., Herbach, U., Richard, A., Guillemin, A., Gonin-Giraud, S., Gros, P.-A., Gandrillon, O., 2019. WASABI: a dynamic iterative framework for gene regulatory network inference. BMC Bioinformatics 20, 220. https://doi.org/10.1186/s12859-019-2798-1   
Bonneau, R., Reiss, D.J., Shannon, P., Facciotti, M., Hood, L., Baliga, N.S., Thorsson, V., 2006. The Inferelator: an algorithm for learning parsimonious regulatory networks from systems-biology data sets de novo. Genome Biol. 7, R36. https://doi.org/10.1186/gb-2006-7-5-r36   
Breiman, L., 2001. Random Forests. Mach. Learn. 45, 5–32. https://doi.org/10.1023/A:1010933404324   
Chan, T.E., Stumpf, M.P.H., Babtie, A.C., 2017. Gene Regulatory Network Inference from Single-Cell Data Using Multivariate Information Measures. Cell Syst. 5, 251-267.e3. https://doi.org/10.1016/j.cels.2017.08.014   
Chen, H., Mundra, P.A., Zhao, L.N., Lin, F., Zheng, J., 2014. Highly sensitive inference of time-delayed gene regulation by network deconvolution. BMC Syst. Biol. 8, S6. https://doi.org/10.1186/1752-0509-8-S4-S6   
Chen, S., Mar, J.C., 2018. Evaluating methods of inferring gene regulatory networks highlights their lack of performance for single cell gene expression data. BMC Bioinformatics 19, 232. https://doi.org/10.1186/s12859-018-2217-z   
Deshpande, A., Chu, L.-F., Stewart, R., Gitter, A., 2022. Network inference with Granger causality ensembles on single-cell transcriptomics. Cell Rep. 38, 110333. https://doi.org/10.1016/j.celrep.2022.110333   
Duren, Z., Lu, W.S., Arthur, J.G., Shah, P., Xin, J., Meschi, F., Li, M.L., Nemec, C.M., Yin, Y., Wong, W.H., 2021. Sc-compReg enables the comparison of gene regulatory networks between conditions using single-cell data. Nat. Commun. 12, 4763. https://doi.org/10.1038/s41467-021-25089-2   
Emmert-Streib, F., Dehmer, M., Haibe-Kains, B., 2014. Gene regulatory networks and their applications: understanding biological and medical problems in terms of networks. Front. Cell Dev. Biol. 2.   
Faith, J.J., Hayete, B., Thaden, J.T., Mogno, I., Wierzbowski, J., Cottarel, G., Kasif, S., Collins, J.J., Gardner, T.S., 2007. Large-Scale Mapping and Validation of Escherichia coli Transcriptional Regulation from Compendium of Expression Profiles. PLOS Biol. 5, e8. https://doi.org/10.1371/journal.pbio.0050008   
Feizi, S., Marbach, D., Médard, M., Kellis, M., 2013. Network deconvolution as a general method to distinguish direct dependencies in networks. Nat. Biotechnol. 31, 726–733. https://doi.org/10.1038/nbt.2635   
Fiers, M.W.E.J., Minnoye, L., Aibar, S., Bravo González-Blas, C., Kalender Atak, Z., Aerts, S., 2018. Mapping gene regulatory networks from single-cell omics data. Brief. Funct. Genomics 17, 246–254. https://doi.org/10.1093/bfgp/elx046   
Finkle, J.D., Wu, J.J., Bagheri, N., 2018. Windowed Granger causal inference strategy improves discovery of gene regulatory networks. Proc. Natl. Acad. Sci. 115, 2252–2257. https://doi.org/10.1073/pnas.1710936115   
Friedman, N., Koller, D., 2003. Being Bayesian About Network Structure. A Bayesian Approach to Structure Discovery in Bayesian Networks. Mach. Learn. 50, 95–125. https://doi.org/10.1023/A:1020249912095   
Fuller, T.F., Ghazalpour, A., Aten, J.E., Drake, T.A., Lusis, A.J., Horvath, S., 2007. Weighted gene coexpression network analysis strategies applied to mouse weight. Mamm. Genome 18, 463–472. https://doi.org/10.1007/s00335-007-9043-3   
Gambardella, G., Moretti, M.N., de Cegli, R., Cardone, L., Peron, A., di Bernardo, D., 2013. Differential network analysis for the identification of condition-specific pathway activity and regulation. Bioinformatics 29, 1776–1785. https://doi.org/10.1093/bioinformatics/btt290   
Gjerga, E., Trairatphisan, P., Gabor, A., Koch, H., Chevalier, C., Ceccarelli, F., Dugourd, A., Mitsos, A., Saez-Rodriguez, J., 2020. Converting networks to predictive logic models from perturbation signalling data with CellNOpt. Bioinformatics 36, 4523–4524. https://doi.org/10.1093/bioinformatics/btaa561   
Glass, L., Kauffman, S.A., 1973. The logical analysis of continuous, non-linear biochemical control networks. J. Theor. Biol. 39, 103–129. https://doi.org/10.1016/0022-5193(73)90208-7   
Grau, J., Grosse, I., Keilwagen, J., 2015. PRROC: computing and visualizing precision-recall and receiver operating characteristic curves in R. Bioinformatics 31, 2595–2597. https://doi.org/10.1093/bioinformatics/btv153   
Hall, Niarakis, 2021. Data integration in logic-based models of biological mechanisms | Elsevier Enhanced Reader [WWW Document]. https://doi.org/10.1016/j.coisb.2021.100386   
Hamey, F.K., Nestorowa, S., Kinston, S.J., Kent, D.G., Wilson, N.K., Göttgens, B., 2017. Reconstructing blood stem cell regulatory network models from single-cell molecular profiles. Proc. Natl. Acad. Sci. 114, 5822–5829. https://doi.org/10.1073/pnas.1610609114   
He, Q., Macauley, M., Davies, R., 2015. Chapter 5 - Dynamics of Complex Boolean Networks: Canalization, Stability, and Criticality, in: Robeva, R.S. (Ed.), Algebraic and Discrete Mathematical Methods for Modern Biology. Academic Press, Boston, pp. 93–119. https://doi.org/10.1016/B978-0-12-801213-0.00005-8   
Henao, J.D., Lauber, M., Azevedo, M., Grekova, A., List, M., Ogris, C., Schubert, B., 2022. Multi-Omics Regulatory Network Inference in the Presence of Missing Data (preprint). Systems Biology. https://doi.org/10.1101/2022.04.14.488153   
Herbach, U., Bonnaffoux, A., Espinasse, T., Gandrillon, O., 2017. Inferring gene regulatory networks from single-cell data: a mechanistic approach. BMC Syst. Biol. 11, 105. https://doi.org/10.1186/s12918-017-0487-0   
Hossain, S.M.M., Halsana, A.A., Khatun, L., Ray, S., Mukhopadhyay, A., 2021. Discovering key transcriptomic regulators in pancreatic ductal adenocarcinoma using Dirichlet process Gaussian mixture model. Sci. Rep. 11, 7853. https://doi.org/10.1038/s41598-021-87234-7   
Hu, Y., Peng, T., Gao, L., Tan, K., 2021. CytoTalk: De novo construction of signal transduction networks using single-cell transcriptomic data. Sci. Adv. 7, eabf1356. https://doi.org/10.1126/sciadv.abf1356   
Huynh-Thu, V.A., Geurts, P., 2018. dynGENIE3: dynamical GENIE3 for the inference of gene networks from time series expression data. Sci. Rep. 8, 3384. https://doi.org/10.1038/s41598-018-21715-0   
Huynh-Thu, V.A., Irrthum, A., Wehenkel, L., Geurts, P., 2010. Inferring Regulatory Networks from Expression Data Using Tree-Based Methods. PLOS ONE 5, e12776. https://doi.org/10.1371/journal.pone.0012776   
Huynh-Thu, V.A., Sanguinetti, G., 2015. Combining tree-based and dynamical systems for the inference of gene regulatory networks. Bioinforma. Oxf. Engl. 31, 1614–1622. https://doi.org/10.1093/bioinformatics/btu863   
Iglesias-Martinez, L.F., Kolch, W., Santra, T., 2016. BGRMI: A method for inferring gene regulatory networks from time-course gene expression data and its application in breast cancer research. Sci. Rep. 6, 37140. https://doi.org/10.1038/srep37140   
Jackson, C.A., Castro, D.M., Saldi, G.-A., Bonneau, R., Gresham, D., 2020. Gene regulatory network reconstruction using single-cell RNA sequencing of barcoded genotypes in diverse environments. eLife 9, e51254. https://doi.org/10.7554/eLife.51254   
Janssens, J., Aibar, S., Taskiran, I.I., Ismail, J.N., Gomez, A.E., Aughey, G., Spanier, K.I., De Rop, F.V., González-Blas, C.B., Dionne, M., Grimes, K., Quan, X.J., Papasokrati, D., Hulselmans, G., Makhzami, S., De Waegeneer, M., Christiaens, V., Southall, T., Aerts, S., 2022. Decoding gene regulation in the fly brain. Nature 601, 630–636. https://doi.org/10.1038/s41586-021-04262-z   
Jungck, J.R., Viswanathan, R., 2015. Chapter 1 - Graph Theory for Systems Biology: Interval Graphs, Motifs, and Pattern Recognition, in: Robeva, R.S. (Ed.), Algebraic and Discrete Mathematical Methods for Modern Biology. Academic Press, Boston, pp. 1–27. https://doi.org/10.1016/B978-0-12-801213-0.00001-0   
Kanehisa, M., Araki, M., Goto, S., Hattori, M., Hirakawa, M., Itoh, M., Katayama, T., Kawashima, S., Okuda, S., Tokimatsu, T., Yamanishi, Y., 2008. KEGG for linking genomes to life and the environment. Nucleic Acids Res. 36, D480–D484. https://doi.org/10.1093/nar/gkm882   
Kang, Y., Thieffry, D., Cantini, L., 2021. Evaluating the Reproducibility of Single-Cell Gene Regulatory Network Inference Algorithms. Front. Genet. 12. https://doi.org/10.3389/fgene.2021.617282   
Kauffman, S.A., Kauffman, M. of the S.F.I. and P. of B.S.A., 1993. The Origins of Order: Self-organization and Selection in Evolution. Oxford University Press.   
Kim, S., 2015. ppcor: An R Package for a Fast Calculation to Semi-partial Correlation Coefficients. Commun. Stat. Appl. Methods 22, 665–674. https://doi.org/10.5351/CSAM.2015.22.6.665   
Kim, S., Kim, J., Cho, K.-H., 2007. Inferring gene regulatory networks from temporal expression profiles under time-delay and noise. Comput. Biol. Chem. 31, 239–245. https://doi.org/10.1016/j.compbiolchem.2007.03.013   
Kimura, S., Fukutomi, R., Tokuhisa, M., Okada, M., 2020. Inference of Genetic Networks From Time-Series and Static Gene Expression Data: Combining a Random-Forest-Based Inference Method With Feature Selection Methods. Front. Genet. 11.   
Klamt, S., Saez-Rodriguez, J., Lindquist, J.A., Simeoni, L., Gilles, E.D., 2006. A methodology for the structural and functional analysis of signaling and regulatory networks. BMC Bioinformatics 7, 56. https://doi.org/10.1186/1471-2105-7-56   
Koh, G.C.K.W., Porras, P., Aranda, B., Hermjakob, H., Orchard, S.E., 2012. Analyzing Protein–Protein Interaction Networks. J. Proteome Res. 11, 2014–2031. https://doi.org/10.1021/pr201211w   
Kolch, W., Halasz, M., Granovskaya, M., Kholodenko, B.N., 2015. The dynamic control of signal transduction networks in cancer cells. Nat. Rev. Cancer 15, 515–527. https://doi.org/10.1038/nrc3983   
Kolpakov, F.A., Ananko, E.A., Kolesov, G.B., Kolchanov, N.A., 1998. GeneNet: a gene network database and its automated visualization. Bioinformatics 14, 529–537. https://doi.org/10.1093/bioinformatics/14.6.529   
Lähdesmäki, H., Shmulevich, I., Yli-Harja, O., 2003. On Learning Gene Regulatory Networks Under the Boolean Network Model. Mach. Learn. 52, 147–167. https://doi.org/10.1023/A:1023905711304   
Lambrechts, D., Wauters, E., Boeckx, B., Aibar, S., Nittner, D., Burton, O., Bassez, A., Decaluwé, H., Pircher, A., Van den Eynde, K., Weynand, B., Verbeken, E., De Leyn, P., Liston, A., Vansteenkiste, J., Carmeliet, P., Aerts, S., Thienpont, B., 2018. Phenotype molding of stromal cells in the lung tumor microenvironment. Nat. Med. 24, 1277–1289. https://doi.org/10.1038/s41591-018-0096-5   
Lee, D., Cho, K.-H., 2018. Topological estimation of signal flow in complex signaling networks. Sci. Rep. 8, 5262. https://doi.org/10.1038/s41598-018-23643-5   
Letort, G., Montagud, A., Stoll, G., Heiland, R., Barillot, E., Macklin, P., Zinovyev, A., Calzone, L., 2019. PhysiBoSS: a multi-scale agent-based modelling framework integrating physical dimension and cell signalling. Bioinformatics 35, 1188–1196. https://doi.org/10.1093/bioinformatics/bty766   
Liang, S., Fuhrman, S., Somogyi, R., 1998. Reveal, a general reverse engineering algorithm for inference of genetic network architectures. Pac. Symp. Biocomput. Pac. Symp. Biocomput. 18–29.   
Lichtblau, Y., Zimmermann, K., Haldemann, B., Lenze, D., Hummel, M., Leser, U., 2017. Comparative assessment of differential network analysis methods. Brief. Bioinform. 18, 837–850. https://doi.org/10.1093/bib/bbw061   
Liu, W., Zhu, W., Liao, B., Chen, H., Ren, S., Cai, L., 2017. Improving gene regulatory network structure using redundancy reduction in the MRNET algorithm. RSC Adv. 7, 23222–23233. https://doi.org/10.1039/C7RA01557G   
Lu, J., Dumitrascu, B., McDowell, I.C., Jo, B., Barrera, A., Hong, L.K., Leichter, S.M., Reddy, T.E., Engelhardt, B.E., 2021. Causal network inference from gene transcriptional time-series response to glucocorticoids. PLOS Comput. Biol. 17, e1008223. https://doi.org/10.1371/journal.pcbi.1008223   
Luscombe, N.M., Madan Babu, M., Yu, H., Snyder, M., Teichmann, S.A., Gerstein, M., 2004. Genomic analysis of regulatory network dynamics reveals large topological changes. Nature 431, 308–312. https://doi.org/10.1038/nature02782   
Manica, M., Bunne, C., Mathis, R., Cadow, J., Ahsen, M.E., Stolovitzky, G.A., Martínez, M.R., 2021. COSIFER: a Python package for the consensus inference of molecular interaction networks. Bioinformatics 37, 2070–2072. https://doi.org/10.1093/bioinformatics/btaa942   
Marazzi, L., Shah, M., Balakrishnan, S., Patil, A., Vera-Licona, P., 2022. NETISCE: A Network-Based Tool for Cell Fate Reprogramming. bioRxiv. https://doi.org/10.1101/2021.12.30.474582   
Marbach, D., Costello, J.C., Küffner, R., Vega, N.M., Prill, R.J., Camacho, D.M., Allison, K.R., DREAM5 Consortium, Kellis, M., Collins, J.J., Stolovitzky, G., 2012. Wisdom of crowds for robust gene network inference. Nat. Methods 9, 796–804. https://doi.org/10.1038/nmeth.2016   
Margolin, A.A., Nemenman, I., Basso, K., Wiggins, C., Stolovitzky, G., Favera, R.D., Califano, A., 2006. ARACNE: An Algorithm for the Reconstruction of Gene Regulatory Networks in a Mammalian Cellular Context. BMC Bioinformatics 7, S7. https://doi.org/10.1186/1471-2105-7-S1-S7   
Matsumoto, H., Kiryu, H., Furusawa, C., Ko, M.S.H., Ko, S.B.H., Gouda, N., Hayashi, T., Nikaido, I., 2017. SCODE: an efficient regulatory network inference algorithm from single-cell RNA-Seq during differentiation. Bioinformatics 33, 2314–2321. https://doi.org/10.1093/bioinformatics/btx194   
Meyer, P.E., Kontos, K., Lafitte, F., Bontempi, G., 2007. Information-Theoretic Inference of Large Transcriptional Regulatory Networks. EURASIP J. Bioinforma. Syst. Biol. 2007, 1–9. https://doi.org/10.1155/2007/79879   
Michailidis, G., d’Alché-Buc, F., 2013. Autoregressive models for gene regulatory network inference: Sparsity, stability and causality issues. Math. Biosci. 246, 326–334. https://doi.org/10.1016/j.mbs.2013.10.003   
Moerman, T., Aibar Santos, S., Bravo González-Blas, C., Simm, J., Moreau, Y., Aerts, J., Aerts, S., 2019. GRNBoost2 and Arboreto: efficient and scalable inference of gene regulatory networks. Bioinformatics 35, 2159–2161. https://doi.org/10.1093/bioinformatics/bty916   
Montagud, A., Béal, J., Tobalina, L., Traynard, P., Subramanian, V., Szalai, B., Alföldi, R., Puskás, L Valencia, A., Barillot, E., Saez-Rodriguez, J., Calzone, L., 2022. Patient-specific Boolean models of signalling networks guide personalised treatments. eLife 11, e72626. https://doi.org/10.7554/eLife.72626   
Murata, T., 1989. Petri nets: Properties, analysis and applications. Proc. IEEE 77, 541–580. https://doi.org/10.1109/5.24143   
Murphy, K., Mian, S., 1999. Modelling gene expression data using dynamic bayesian networks.   
Müssel, C., Hopfensitz, M., Kestler, H.A., 2010. BoolNet—an R package for generation, reconstruction and analysis of Boolean networks. Bioinformatics 26, 1378–1380. https://doi.org/10.1093/bioinformatics/btq124   
Naldi, A., Hernandez, C., Levy, N., Stoll, G., Monteiro, P.T., Chaouiya, C., Helikar, T., Zinovyev, A., Calzone, L., Cohen-Boulakia, S., Thieffry, D., Paulevé, L., 2018. The CoLoMoTo Interactive Notebook: Accessible and Reproducible Computational Analyses for Qualitative Biological Networks. Front. Physiol. 9.   
Nguyen, H., Tran, D., Tran, B., Pehlivan, B., Nguyen, T., 2021. A comprehensive survey of regulatory network inference methods using single cell RNA sequencing data. Brief. Bioinform. 22, bbaa190. https://doi.org/10.1093/bib/bbaa190   
Ocone, A., Haghverdi, L., Mueller, N.S., Theis, F.J., 2015. Reconstructing gene regulatory dynamics from high-dimensional single-cell snapshot data. Bioinformatics 31, i89–i96. https://doi.org/10.1093/bioinformatics/btv257   
Odibat, O., Reddy, C.K., 2011. Ranking differential genes in co-expression networks, in: Proceedings of the 2nd ACM Conference on Bioinformatics, Computational Biology and Biomedicine, BCB ’11. Association for Computing Machinery, New York, NY, USA, pp. 350–354. https://doi.org/10.1145/2147805.2147849   
Ostrowski, M., Paulevé, L., Schaub, T., Siegel, A., Guziolowski, C., 2016. Boolean network identification from perturbation time series data combining dynamics abstraction and logic programming. Biosystems, Selected papers from the Computational Methods in Systems Biology 2015 conference 149, 139–153. https://doi.org/10.1016/j.biosystems.2016.07.009   
Papili Gao, N., Ud-Dean, S.M.M., Gandrillon, O. Gunawan, R., 2018. SINCERITIES: inferring gene regulatory networks from single cell transcriptional expression profiles. Bioinformatics 34, 258–266. http 10.1093/bioinformatics/btx575   
Park, S., Kim, J.M., Shin, W., Han, S.W., Jeon, M., Jang, H.J., Jang, I.-S., Kang, J., 2018. BTNET : boosted tree based gene regulatory network inference algorithm using time-course measurement data. BMC Syst. Biol. 12, 20. https://doi.org/10.1186/s12918-018-0547-0   
Patsalos, A., Halasz, L., Medina-Serpas, M.A., Berger, W.K., Daniel, B., Tzerpos, P., Kiss, M., Nagy, G., Fischer, C., Simandi, Z., Varga, T., Nagy, L., 2021. A growth factor–expressing macrophage subpopulation orchestrates regenerative inflammation via GDF-15. J. Exp. Med. 219, e20210420. https://doi.org/10.1084/jem.20210420   
Pellegrini, M., Haynor, D., Johnson, J.M., 2004. Protein Interaction Networks. Expert Rev. Proteomics 1, 239–249. https://doi.org/10.1586/14789450.1.2.239   
Petralia, F., Wang, P., Yang, J., Tu, Z., 2015. Integrative random forest for gene regulatory network inference. Bioinformatics 31, i197–i205. https://doi.org/10.1093/bioinformatics/btv268   
Polynikis, A., Hogan, S.J., di Bernardo, M., 2009. Comparing different ODE modelling approaches for gene regulatory networks. J. Theor. Biol. 261, 511–530. https://doi.org/10.1016/j.jtbi.2009.07.040   
Pratapa, A., Jalihal, A.P., Law, J.N., Bharadwaj, A., Murali, T.M., 2020. Benchmarking algorithms for gene regulatory network inference from single-cell transcriptomic data. Nat. Methods 17, 147–154. https://doi.org/10.1038/s41592-019-0690-6   
Pušnik, Ž., Mraz, M., Zimic, N., Moškon, M., 2022. Review and assessment of Boolean approaches for inference of gene regulatory networks. Heliyon 8, e10222. https://doi.org/10.1016/j.heliyon.2022.e10222   
Qiu, X., Rahimzamani, A., Wang, L., Ren, B., Mao, Q., Durham, T., McFaline-Figueroa, J.L., Saunders, L., Trapnell, C., Kannan, S., 2020. Inferring Causal Gene Regulatory Networks from Coupled Single-Cell Expression Dynamics Using Scribe. Cell Syst. 10, 265-274.e11. https://doi.org/10.1016/j.cels.2020.02.003   
Raharinirina, N.A., Peppert, F., von Kleist, M., Schütte, C., Sunkara, V., 2021a. Inferring gene regulatory networks from single-cell RNA-seq temporal snapshot data requires higher-order moments. Patterns 2, 100332. https://doi.org/10.1016/j.patter.2021.100332   
Raharinirina, N.A., Peppert, F., von Kleist, M., Schütte, C., Sunkara, V., 2021b. Inferring gene regulatory networks from single-cell RNA-seq temporal snapshot data requires higher-order moments. Patterns 2, 100332. https://doi.org/10.1016/j.patter.2021.100332   
Razzaq, M., Paulevé, L., Siegel, A., Saez-Rodriguez, J., Bourdon, J., Guziolowski, C., 2018. Computational discovery of dynamic cell line specific Boolean networks from multiplex time-course data. PLOS Comput. Biol. 14, e1006538. https://doi.org/10.1371/journal.pcbi.1006538   
Salgado, H., Peralta-Gil, M., Gama-Castro, S., Santos-Zavaleta, A., Muñiz-Rascado, L., García-Sotelo, J.S., Weiss, V., Solano-Lira, H., Martínez-Flores, I., Medina-Rivera, A., Salgado-Osorio, G., Alquicira-Hernández, S., Alquicira-Hernández, K., López-Fuentes, A., Porrón-Sotelo, L., Huerta, A.M., Bonavides-Martínez, C., Balderas-Martínez, Y.I., Pannier, L., Olvera, M., Labastida, A., Jiménez-Jacinto, V., Vega-Alvarado, L., del Moral-Chávez, V., Hernández-Alvarez, A., Morett, E., Collado-Vides, J., 2013. RegulonDB v8.0: omics data sets, evolutionary conservation, regulatory phrases, cross-validated gold standards and more. Nucleic Acids Res. 41, D203–D213. https://doi.org/10.1093/nar/gks1201   
Sanchez-Castillo, M., Blanco, D., Tienda-Luna, I.M., Carrion, M.C., Huang, Y., 2018. A Bayesian framework for the inference of gene regulatory networks from time and pseudo-time series data. Bioinformatics 34, 964–970. https://doi.org/10.1093/bioinformatics/btx605   
Schwikowski, B., Uetz, P., Fields, S., 2000. A network of protein–protein interactions in yeast. Nat. Biotechnol. 18, 1257–1261. https://doi.org/10.1038/82360   
Shah, O.S., Chaudhary, M.F.A., Awan, H.A., Fatima, F., Arshad, Z., Amina, B., Ahmed, M., Hameed, H., Furqan, M., Khalid, S., Faisal, A., Chaudhary, S.U., 2018. ATLANTIS - Attractor Landscape Analysis Toolbox for Cell Fate Discovery and Reprogramming. Sci. Rep. 8, 3554. https://doi.org/10.1038/s41598-018-22031-3   
Shi, N., Zhu, Z., Tang, K., Parker, D., He, S., 2019. ATEN: And/Or tree ensemble for inferring accurate Boolean network topology and dynamics. Bioinformatics btz563. https://doi.org/10.1093/bioinformatics/btz563   
Shmulevich, I., Dougherty, E.R., Kim, S., Zhang, W., 2002a. Probabilistic Boolean networks: a rule-based uncertainty model for gene regulatory networks. Bioinformatics 18, 261–274. https://doi.org/10.1093/bioinformatics/18.2.261   
Shmulevich, I., Saarinen, A., Yli-Harja, O., Astola, J., 2002b. Inference of Genetic Regulatory Networks Via Best-Fit Extensions, in: Zhang, W., Shmulevich, I. (Eds.), Computational and Statistical Approaches to Genomics. Springer US, Boston, MA, pp. 197–210. https://doi.org/10.1007/0-306-47825-0_11   
Shojaie, A., 2021. Differential network analysis: A statistical perspective. WIREs Comput. Stat. 13, e1508. https://doi.org/10.1002/wics.1508   
Silverman, E.K., Schmidt, H.H.H.W., Anastasiadou, E., Altucci, L., Angelini, M., Badimon, L., Balligand, J.-L., Benincasa, G., Capasso, G., Conte, F., Di Costanzo, A., Farina, L., Fiscon, G., Gatto, L., Gentili, M., Loscalzo, J., Marchese, C., Napoli, C., Paci, P., Petti, M., Quackenbush, J., Tieri, P., Viggiano, D., Vilahur, G., Glass, K., Baumbach, J., 2020. Molecular networks in Network Medicine: Development and applications. Wiley Interdiscip. Rev. Syst. Biol. Med. 12, e1489. https://doi.org/10.1002/wsbm.1489   
Sonawane, A.R., DeMeo, D.L., Quackenbush, J., Glass, K., 2021. Constructing gene regulatory networks using epigenetic data. Npj Syst. Biol. Appl. 7, 45. https://doi.org/10.1038/s41540-021-00208-3   
Sonawane, A.R., Weiss, S.T., Glass, K., Sharma, A., 2019. Network Medicine in the Age of Biomedical Big Data. Front. Genet. 10.   
Specht, A.T., Li, J., 2016. LEAP: constructing gene co-expression networks for single-cell RNA-sequencing data using pseudotime ordering. Bioinformatics btw729. https://doi.org/10.1093/bioinformatics/btw729   
Stoll, G., Naldi, A., Noël, V., Viara, E., Barillot, E., Kroemer, G., Thieffry, D., Calzone, L., 2022. UPMaBoSS: A Novel Framework for Dynamic Cell Population Modeling. Front. Mol. Biosci. 9.   
Takeuchi, Y., 1996. Global Dynamical Properties Of Lotka-volterra Systems. World Scientific.   
Tam, G.H.F., Chang, C., Hung, Y.S., 2013. Gene regulatory network discovery using pairwise Granger causality. IET Syst. Biol. 7, 195–204. https://doi.org/10.1049/iet-syb.2012.0063   
Van Hove, H., Martens, L., Scheyltjens, I., De Vlaminck, K., Pombo Antunes, A.R., De Prijck, S., Vandamme, N., De Schepper, S., Van Isterdael, G., Scott, C.L., Aerts, J., Berx, G., Boeckxstaens, G.E., Vandenbroucke, R.E., Vereecke, L., Moechars, D., Guilliams, M., Van Ginderachter, J.A., Saeys, Y., Movahedi, K., 2019. A single-cell atlas of mouse brain macrophages reveals unique transcriptional identities shaped by ontogeny and tissue environment. Nat. Neurosci. 22, 1021–1035. https://doi.org/10.1038/s41593-019-0393-4   
Vazquez, A., Flammini, A., Maritan, A., Vespignani, A., 2003. Global protein function prediction from protein-protein interaction networks. Nat. Biotechnol. 21, 697–700. https://doi.org/10.1038/nbt825   
Ventre, E., Herbach, U., Espinasse, T., Benoit, G., Gandrillon, O., 2022. One model fits all: combining inference and simulation of gene regulatory networks (preprint). Systems Biology. https://doi.org/10.1101/2022.06.19.496754   
Vijesh, N., Chakrabarti, S.K., Sreekumar, J., 2013. Modeling of gene regulatory networks: A review. J.   
Xing, L., Guo, M., Liu, X., Wang, C., Wang, L., Zhang, Y., 2017. An improved Bayesian network method for reconstructing gene regulatory network based on candidate auto selection. BMC Genomics 18, 844. https://doi.org/10.1186/s12864-017-4228-y   
Xu, H., Baroukh, C., Dannenfelser, R., Chen, E.Y., Tan, C.M., Kou, Y., Kim, Y.E., Lemischka, I.R., Ma’ayan, A., 2013. ESCAPE: database for integrating high-content published data collected from human and mouse embryonic stem cells. Database 2013, bat045. https://doi.org/10.1093/database/bat045   
Yang, G., Gómez Tejeda Zañudo, J., Albert, R., 2018. Target Control in Logical Models Using the Domain of Influence of Nodes. Front. Physiol. 9.   
Yao, S., Yoo, S., Yu, D., 2015. Prior knowledge driven Granger causality analysis on gene regulatory network discovery. BMC Bioinformatics 16, 273. https://doi.org/10.1186/s12859-015-0710-1   
Young, W.C., Raftery, A.E., Yeung, K.Y., 2014. Fast Bayesian inference for gene regulatory networks using ScanBMA. BMC Syst. Biol. 8, 47. https://doi.org/10.1186/1752-0509-8-47   
Zeng, C., Mulas, F., Sui, Y., Guan, T., Miller, N., Tan, Y., Liu, F., Jin, W., Carrano, A.C., Huising, M.O., Shirihai, O.S., Yeo, G.W., Sander, M., 2017. Pseudotemporal ordering of single cells reveals metabolic control of postnatal beta-cell proliferation. Cell Metab. 25, 1160-1175.e11. https://doi.org/10.1016/j.cmet.2017.04.014   
Zoppoli, P., Morganella, S., Ceccarelli, M., 2010. TimeDelay-ARACNE: Reverse engineering of gene networks from time-course data by an information theoretic approach. BMC Bioinformatics 11, 154. https://doi.org/10.1186/1471-2105-11-154  